{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Author :** Rutendo F. Sigauke\n",
    "\n",
    "\n",
    "# Summary\n",
    "\n",
    "- Generate updated normalized counts\n",
    "\n",
    "- Take the opposite sense strand of gene counts if a bidirectional transcript overlaps a gene. This includes 15kb downstream on PAS and upstream of TSS by 250bp.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table) ##fast loading of large tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions for processing \n",
    "\n",
    "## TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "tpm_calculation <- function(count_g,len_g) {\n",
    "    \n",
    "    #' calculate tpms for counts #https://support.bioconductor.org/p/91218/\n",
    "    #' \n",
    "    #' @description This function will calculate tpms for counts table\n",
    "    #' derived from the same library.  \n",
    "    #' \n",
    "    #' @param count_g data.frame. first counts\n",
    "    #' \n",
    "    #' @param len_g list of integers. Transcript lengths corresponding to count_g\n",
    "    #' \n",
    "    #'\n",
    "    #' @usage tpm(count_g, len_g)\n",
    "    #' @return Matix with tpms values for genes as a data.frame\n",
    "    #' @export\n",
    "\n",
    "    #get length normalized counts\n",
    "    g <- count_g/len_g\n",
    "\n",
    "    #get sum of length normalized counts\n",
    "    tot_tpms <- colSums(g)\n",
    "\n",
    "    #calculate the TPMs\n",
    "    tpm_g <- t(t(g)*1e6/tot_tpms)\n",
    "\n",
    "    return(tpm_g)   \n",
    "  \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get strand counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_strand_counts <- function(counts, strands, strand_name){\n",
    "    \n",
    "    #subset bidirectionals based on the genes' strand\n",
    "    strands_filts <- subset(strands, gene_strand==strand_name)\n",
    "    \n",
    "    #get the counts from the filtered counts\n",
    "    counts_updated <- counts[counts$bidirs %in% strands_filts$bidirs,]\n",
    "    return(counts_updated)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Coefficient of variation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cv <- function(x){\n",
    "    #' calculate coefficient of variation for set of values\n",
    "    #' \n",
    "    #' @description This function will calculate coefficient of variation\n",
    "    #' \n",
    "    #' @param x list of values\n",
    "    #' \n",
    "    #' @usage cv(x)\n",
    "    #' @return CV value\n",
    "    #' @export\n",
    "    \n",
    "    #sd(x, na.rm=TRUE)/mean(x, na.rm=TRUE)\n",
    "    sd(x, na.rm=FALSE)/mean(x, na.rm=FALSE)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Summary of transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_summary_statistics <- function(counts_df, genenames, bidirnames){\n",
    "    \n",
    "    #1. transcript lenths\n",
    "    transcript_lengths <- (counts_df$stop - counts_df$start) +1\n",
    "\n",
    "    #2. number of columns in the dataframes\n",
    "    ncolumns <- ncol(counts_df)\n",
    "\n",
    "    \n",
    "    ##get summary statistics for bed6+counts data.frame\n",
    "    #3. transcript coordinated in bed6 format\n",
    "    df_summary <- counts_df[,1:6]\n",
    "\n",
    "    #4. get stats like length, st.dev, etc\n",
    "    df_summary$length <- transcript_lengths\n",
    "    df_summary$stdev <- apply(counts_df[c(7:ncolumns)], 1, sd)\n",
    "    df_summary$variance <- apply(counts_df[c(7:ncolumns)], 1, var)\n",
    "    df_summary$sum <- apply(counts_df[c(7:ncolumns)], 1, sum)\n",
    "    df_summary$mean <- apply(counts_df[c(7:ncolumns)], 1, mean)\n",
    "    df_summary$median <- apply(counts_df[c(7:ncolumns)], 1, median)\n",
    "    df_summary$coefvar <- apply(counts_df[c(7:ncolumns)], 1, cv)\n",
    "    df_summary$max <- apply(counts_df[c(7:ncolumns)], 1, max)\n",
    "    df_summary$min <- apply(counts_df[c(7:ncolumns)], 1, min)\n",
    "    df_summary$num_transcribed <- rowSums(counts_df[c(7:ncolumns)] != 0)\n",
    "    df_summary$percent_transcribed <- (df_summary$num_transcribed/(ncolumns-6))*100\n",
    "    df_summary$transcript_type <- ifelse(df_summary$gene_transcript %in% genenames,\n",
    "                                                 \"genes\",\n",
    "                                                 ifelse(df_summary$gene_transcript %in% bidirnames,\n",
    "                                                        \"bidirectional\",\"NA\"))\n",
    "    return(df_summary)\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## QC filtered samples\n",
    "\n",
    "These samples were filtered in the dbNascent paper based on GC content of sample and sample QC."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "751"
      ],
      "text/latex": [
       "751"
      ],
      "text/markdown": [
       "751"
      ],
      "text/plain": [
       "[1] 751"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 14</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>organism</th><th scope=col>tissue</th><th scope=col>tissue_description</th><th scope=col>disease</th><th scope=col>srp</th><th scope=col>paper_id</th><th scope=col>protocol</th><th scope=col>sample_name</th><th scope=col>sample_type</th><th scope=col>cell_type</th><th scope=col>samp_qc_score</th><th scope=col>samp_data_score</th><th scope=col>replicate</th><th scope=col>control_experimental</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>H. sapiens</td><td>blood</td><td>blood; biphenotypic B myelomonocytic leukemia</td><td>1</td><td>SRP149838</td><td>Aho2019displacement</td><td>PRO-seq</td><td>SRR7266931</td><td>cell line</td><td>MV4-11</td><td>1</td><td>2</td><td>1</td><td>control     </td></tr>\n",
       "\t<tr><td>H. sapiens</td><td>blood</td><td>blood; biphenotypic B myelomonocytic leukemia</td><td>1</td><td>SRP149838</td><td>Aho2019displacement</td><td>PRO-seq</td><td>SRR7266932</td><td>cell line</td><td>MV4-11</td><td>2</td><td>1</td><td>2</td><td>control     </td></tr>\n",
       "\t<tr><td>H. sapiens</td><td>blood</td><td>blood; biphenotypic B myelomonocytic leukemia</td><td>1</td><td>SRP149838</td><td>Aho2019displacement</td><td>PRO-seq</td><td>SRR7266933</td><td>cell line</td><td>MV4-11</td><td>1</td><td>2</td><td>1</td><td>experimental</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 14\n",
       "\\begin{tabular}{llllllllllllll}\n",
       " organism & tissue & tissue\\_description & disease & srp & paper\\_id & protocol & sample\\_name & sample\\_type & cell\\_type & samp\\_qc\\_score & samp\\_data\\_score & replicate & control\\_experimental\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t H. sapiens & blood & blood; biphenotypic B myelomonocytic leukemia & 1 & SRP149838 & Aho2019displacement & PRO-seq & SRR7266931 & cell line & MV4-11 & 1 & 2 & 1 & control     \\\\\n",
       "\t H. sapiens & blood & blood; biphenotypic B myelomonocytic leukemia & 1 & SRP149838 & Aho2019displacement & PRO-seq & SRR7266932 & cell line & MV4-11 & 2 & 1 & 2 & control     \\\\\n",
       "\t H. sapiens & blood & blood; biphenotypic B myelomonocytic leukemia & 1 & SRP149838 & Aho2019displacement & PRO-seq & SRR7266933 & cell line & MV4-11 & 1 & 2 & 1 & experimental\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 14\n",
       "\n",
       "| organism &lt;chr&gt; | tissue &lt;chr&gt; | tissue_description &lt;chr&gt; | disease &lt;int&gt; | srp &lt;chr&gt; | paper_id &lt;chr&gt; | protocol &lt;chr&gt; | sample_name &lt;chr&gt; | sample_type &lt;chr&gt; | cell_type &lt;chr&gt; | samp_qc_score &lt;int&gt; | samp_data_score &lt;int&gt; | replicate &lt;int&gt; | control_experimental &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| H. sapiens | blood | blood; biphenotypic B myelomonocytic leukemia | 1 | SRP149838 | Aho2019displacement | PRO-seq | SRR7266931 | cell line | MV4-11 | 1 | 2 | 1 | control      |\n",
       "| H. sapiens | blood | blood; biphenotypic B myelomonocytic leukemia | 1 | SRP149838 | Aho2019displacement | PRO-seq | SRR7266932 | cell line | MV4-11 | 2 | 1 | 2 | control      |\n",
       "| H. sapiens | blood | blood; biphenotypic B myelomonocytic leukemia | 1 | SRP149838 | Aho2019displacement | PRO-seq | SRR7266933 | cell line | MV4-11 | 1 | 2 | 1 | experimental |\n",
       "\n"
      ],
      "text/plain": [
       "  organism   tissue tissue_description                            disease\n",
       "1 H. sapiens blood  blood; biphenotypic B myelomonocytic leukemia 1      \n",
       "2 H. sapiens blood  blood; biphenotypic B myelomonocytic leukemia 1      \n",
       "3 H. sapiens blood  blood; biphenotypic B myelomonocytic leukemia 1      \n",
       "  srp       paper_id            protocol sample_name sample_type cell_type\n",
       "1 SRP149838 Aho2019displacement PRO-seq  SRR7266931  cell line   MV4-11   \n",
       "2 SRP149838 Aho2019displacement PRO-seq  SRR7266932  cell line   MV4-11   \n",
       "3 SRP149838 Aho2019displacement PRO-seq  SRR7266933  cell line   MV4-11   \n",
       "  samp_qc_score samp_data_score replicate control_experimental\n",
       "1 1             2               1         control             \n",
       "2 2             1               2         control             \n",
       "3 1             2               1         experimental        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "filtered_samples <- data.table::fread(\"/Users/rusi2317/projects/meta_analysis_qc/hg38/processed_data/counts/normalized/human_samples_QC_GC_protocol_filtered.tsv.gz\")\n",
    "nrow(filtered_samples)\n",
    "head(filtered_samples, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene counts\n",
    "\n",
    "- There are `5' Truncated gene annotations` "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>28889</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 28889\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 28889\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 28889  1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>43</td><td>4</td><td>45</td><td>34</td><td>⋯</td><td>12</td><td>10</td><td>38</td><td>30</td><td>42</td><td>62</td><td>53</td><td>44</td><td>49</td><td>61</td></tr>\n",
       "\t<tr><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>chr1</td><td>34610</td><td>35591</td><td>FAM138A:NR_026818.1  </td><td>.</td><td>-</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2 </td><td>.</td><td>+</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2   & . & + &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\t chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1    & . & - & 43 & 4 & 45 & 34 & ⋯ & 12 & 10 & 38 & 30 & 42 & 62 & 53 & 44 & 49 & 61\\\\\n",
       "\t chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\t chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\t chr1 & 34610 & 35591 & FAM138A:NR\\_026818.1   & . & - &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\t chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2  & . & + &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 12623 | 14409 | DDX11L1:NR_046018.2   | . | + |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "| chr1 | 14361 | 28620 | WASH7P:NR_024540.1    | . | - | 43 | 4 | 45 | 34 | ⋯ | 12 | 10 | 38 | 30 | 42 | 62 | 53 | 44 | 49 | 61 |\n",
       "| chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "| chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "| chr1 | 34610 | 35591 | FAM138A:NR_026818.1   | . | - |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "| chr1 | 66168 | 71585 | OR4F5:NM_001005484.2  | . | + |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript       score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2   .     +       0         0         \n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1    .     -      43         4         \n",
       "3 chr1  17368 17436 MIR6859-1:NR_106918.1 .     -       0         0         \n",
       "4 chr1  30365 30503 MIR1302-2:NR_036051.1 .     +       0         0         \n",
       "5 chr1  34610 35591 FAM138A:NR_026818.1   .     -       0         0         \n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2  .     +       0         0         \n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1  0          0         ⋯  0          0          0          0        \n",
       "2 45         34         ⋯ 12         10         38         30        \n",
       "3  0          0         ⋯  0          0          0          0        \n",
       "4  0          0         ⋯  0          0          0          0        \n",
       "5  0          0         ⋯  0          0          0          0        \n",
       "6  0          0         ⋯  0          0          0          0        \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1  0          0          0          0          0          0        \n",
       "2 42         62         53         44         49         61        \n",
       "3  0          0          0          0          0          0        \n",
       "4  0          0          0          0          0          0        \n",
       "5  0          0          0          0          0          0        \n",
       "6  0          0          0          0          0          0        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts_genes <- data.table::fread('/Users/rusi2317/projects/meta_analysis_qc/hg38/processed_data/counts/genes/counts_filt_5ptrunc_gene_stranded_counts.txt',\n",
    "                                 sep='\\t')\n",
    "dim(counts_genes)\n",
    "head(counts_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bidirectional counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>847521</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 847521\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 847521\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 847521   1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand SRR7266931 SRR7266932 SRR7266933\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      0          0          0         \n",
       "2 chr1  5632  6042 chr1:5632-6042 14    .      0          0          0         \n",
       "3 chr1  6132  6486 chr1:6132-6486  7    .      0          0          0         \n",
       "  SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719 SRR8483105\n",
       "1 0          ⋯ 0          0          0          0          0         \n",
       "2 0          ⋯ 0          0          0          0          0         \n",
       "3 0          ⋯ 0          0          0          0          0         \n",
       "  SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0         \n",
       "2 0          0          0          0          0         \n",
       "3 0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts_neg_bidirs <- data.table::fread('/Users/rusi2317/projects/meta_analysis_qc/hg38/processed_data/counts/bidirectionals/counts_bidirectionals_neg.bed',\n",
    "                                  sep='\\t')\n",
    "counts_neg_bidirs$bidirs <- paste0(counts_neg_bidirs$chrom,':',\n",
    "                              counts_neg_bidirs$start,'-',\n",
    "                              counts_neg_bidirs$end)\n",
    "dim(counts_neg_bidirs)\n",
    "head(counts_neg_bidirs, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>847521</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 847521\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 847521\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 847521   1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand SRR7266931 SRR7266932 SRR7266933\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      0          0          0         \n",
       "2 chr1  5632  6042 chr1:5632-6042 14    .      0          0          0         \n",
       "3 chr1  6132  6486 chr1:6132-6486  7    .      0          0          0         \n",
       "  SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719 SRR8483105\n",
       "1 0          ⋯ 0          0          0          0          0         \n",
       "2 0          ⋯ 0          0          0          0          0         \n",
       "3 0          ⋯ 0          0          0          0          0         \n",
       "  SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0         \n",
       "2 0          0          0          0          0         \n",
       "3 0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts_pos_bidirs <- data.table::fread('/Users/rusi2317/projects/meta_analysis_qc/hg38/processed_data/counts/bidirectionals/counts_bidirectionals_pos.bed',\n",
    "                                  sep='\\t')\n",
    "counts_pos_bidirs$bidirs <- paste0(counts_pos_bidirs$chrom,':',\n",
    "                              counts_pos_bidirs$start,'-',\n",
    "                              counts_pos_bidirs$end)\n",
    "dim(counts_pos_bidirs)\n",
    "head(counts_pos_bidirs, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>847521</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 847521\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 847521\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 847521   1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand SRR7266931 SRR7266932 SRR7266933\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      0          0          0         \n",
       "2 chr1  5632  6042 chr1:5632-6042 14    .      0          0          0         \n",
       "3 chr1  6132  6486 chr1:6132-6486  7    .      0          0          0         \n",
       "  SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719 SRR8483105\n",
       "1 0          ⋯ 0          0          0          0          0         \n",
       "2 0          ⋯ 0          0          0          0          0         \n",
       "3 0          ⋯ 0          0          0          0          0         \n",
       "  SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0         \n",
       "2 0          0          0          0          0         \n",
       "3 0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nsamples <- as.numeric(ncol(counts_pos_bidirs))\n",
    "\n",
    "counts_bidirs_pos_neg <- cbind(counts_pos_bidirs[,1:6],\n",
    "                       counts_pos_bidirs[,7:nsamples] + counts_neg_bidirs[,7:nsamples])\n",
    "dim(counts_bidirs_pos_neg)\n",
    "head(counts_bidirs_pos_neg, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Bidirectional filters"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```sh\n",
    "\n",
    "module load bedtools/2.28.0\n",
    "\n",
    "###########################\n",
    "##0: initialize files    ## \n",
    "###########################\n",
    "wd=/scratch/Users/rusi2317/projects/meta_analysis_qc_reviews\n",
    "hg38_refseq=/scratch/Shares/dowell/genomes/hg38/ncbi/hg38_refseq_diff53prime_with_putatives.bed  \n",
    "hg38_bidirs=/scratch/Shares/dowell/dbnascent/out/meta_analysis/mumerge/bidirectionals_dreg_tfit/hg38_tfit_dreg_bidirectionals.bed\n",
    "genome=/scratch/Shares/dowell/genomes/hg38/ncbi/hg38.chrom.sizes_unsorted\n",
    "outdir=${wd}/annotations\n",
    "\n",
    "mkdir -p ${outdir}\n",
    "\n",
    "########################################################################\n",
    "#1: get bidirectionals NOT overlapping RefSeq truncated genes         ##\n",
    "########################################################################\n",
    "\n",
    "#filter RefSeq annotations so that only the curated gene list is used (NM,NR,NP)\n",
    "cat ${hg38_refseq} | grep \"NM_\\|NR_\\|NP_\" > ${outdir}/hg38_refseq_diff53prime_curated.bed\n",
    "\n",
    "#add slop 250 bp upstream of TSS (-l 250) and 15kb from PAS (-r 15000)\n",
    "bedtools slop -i ${hg38_refseq} -g ${genome} -l 250 -r 15000 -s > ${outdir}/hg38_refseq_diff53prime_with_putatives_SLOP.bed\n",
    "\n",
    "#i: get bidirectional transcripts NOT overlapping RefSeq\n",
    "#Takes the above steps and puts them together\n",
    "cat ${hg38_refseq} | grep \"NM_\\|NR_\\|NP_\" | bedtools slop -i - -g ${genome} -l 250 -r 15000 -s | bedtools intersect -a ${hg38_bidirs} -b - -wa -v > ${outdir}/hg38_tfit_dreg_bidirectionals_non_genes.bed\n",
    "\n",
    "#ii: get those that do overlap, along with the gene annotations\n",
    "#filter RefSeq annotations so that only the curated gene list is used (NM,NR,NP)\n",
    "#report the number of bases that overlap\n",
    "cat ${hg38_refseq} | grep \"NM_\\|NR_\\|NP_\" | bedtools slop -i - -g ${genome} -l 250 -r 15000 -s | bedtools intersect -a ${hg38_bidirs} -b - -wo > ${outdir}/hg38_tfit_dreg_bidirectionals_in_genes.bed \n",
    "\n",
    "########################################################################\n",
    "#2: get distances between genes and nearest genes. Used for filtering ##\n",
    "########################################################################\n",
    "#i: get distances between genes (used to filter those with close genes downstream)                                                                                                                                             \n",
    "#D - Use negative distances to report upstream features\n",
    "#a - Report distance with respect to A \n",
    "#-k report upto 2 closest\n",
    "#-S require opposite strandedness\n",
    "bedtools closest -a ${outdir}/hg38_refseq_diff53prime_curated.bed -b ${outdir}/hg38_refseq_diff53prime_curated.bed -S -D a -k 2 > ${outdir}/hg38_refseq_diff53prime_genes_distances.bed\n",
    "\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bidirectional transcripts NOT overlapping genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "258345"
      ],
      "text/latex": [
       "258345"
      ],
      "text/markdown": [
       "258345"
      ],
      "text/plain": [
       "[1] 258345"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>46585</td><td>47033</td><td>chr1:46585-47033</td><td>3</td><td>.</td></tr>\n",
       "\t<tr><td>chr1</td><td>53210</td><td>53580</td><td>chr1:53210-53580</td><td>1</td><td>.</td></tr>\n",
       "\t<tr><td>chr1</td><td>60170</td><td>60510</td><td>chr1:60170-60510</td><td>1</td><td>.</td></tr>\n",
       "\t<tr><td>chr1</td><td>61756</td><td>62124</td><td>chr1:61756-62124</td><td>1</td><td>.</td></tr>\n",
       "\t<tr><td>chr1</td><td>87654</td><td>88246</td><td>chr1:87654-88246</td><td>4</td><td>.</td></tr>\n",
       "\t<tr><td>chr1</td><td>88908</td><td>89232</td><td>chr1:88908-89232</td><td>7</td><td>.</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chrom & start & end & bidirs & score & strand\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr1 & 46585 & 47033 & chr1:46585-47033 & 3 & .\\\\\n",
       "\t chr1 & 53210 & 53580 & chr1:53210-53580 & 1 & .\\\\\n",
       "\t chr1 & 60170 & 60510 & chr1:60170-60510 & 1 & .\\\\\n",
       "\t chr1 & 61756 & 62124 & chr1:61756-62124 & 1 & .\\\\\n",
       "\t chr1 & 87654 & 88246 & chr1:87654-88246 & 4 & .\\\\\n",
       "\t chr1 & 88908 & 89232 & chr1:88908-89232 & 7 & .\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 6\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| chr1 | 46585 | 47033 | chr1:46585-47033 | 3 | . |\n",
       "| chr1 | 53210 | 53580 | chr1:53210-53580 | 1 | . |\n",
       "| chr1 | 60170 | 60510 | chr1:60170-60510 | 1 | . |\n",
       "| chr1 | 61756 | 62124 | chr1:61756-62124 | 1 | . |\n",
       "| chr1 | 87654 | 88246 | chr1:87654-88246 | 4 | . |\n",
       "| chr1 | 88908 | 89232 | chr1:88908-89232 | 7 | . |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end   bidirs           score strand\n",
       "1 chr1  46585 47033 chr1:46585-47033 3     .     \n",
       "2 chr1  53210 53580 chr1:53210-53580 1     .     \n",
       "3 chr1  60170 60510 chr1:60170-60510 1     .     \n",
       "4 chr1  61756 62124 chr1:61756-62124 1     .     \n",
       "5 chr1  87654 88246 chr1:87654-88246 4     .     \n",
       "6 chr1  88908 89232 chr1:88908-89232 7     .     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bidir_no_overlaps <- data.table::fread(\"/Users/rusi2317/projects/meta_analysis_qc_reviews/annotations/hg38_tfit_dreg_bidirectionals_non_genes.bed\",\n",
    "                                   sep=\"\\t\")\n",
    "colnames(bidir_no_overlaps) <- c('chrom','start','end','bidirs',\n",
    "                             'score','strand')\n",
    "bidir_no_overlaps$bidirs <- paste0(bidir_no_overlaps$chrom,':',\n",
    "                              bidir_no_overlaps$start,'-',\n",
    "                              bidir_no_overlaps$end)\n",
    "\n",
    "nrow(bidir_no_overlaps)\n",
    "head(bidir_no_overlaps)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Bidirectional transcripts DO overlap genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "1177667"
      ],
      "text/latex": [
       "1177667"
      ],
      "text/markdown": [
       "1177667"
      ],
      "text/plain": [
       "[1] 1177667"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gene_chrom</th><th scope=col>gene_start</th><th scope=col>gene_end</th><th scope=col>gene_id</th><th scope=col>gene_score</th><th scope=col>gene_strand</th><th scope=col>overlap_len</th><th scope=col>bidir_width</th><th scope=col>bidir_overlap_perc</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>2368</td><td>17686</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td> 410</td><td> 410</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>chr1</td><td>2368</td><td>17686</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td> 410</td><td> 410</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td> 354</td><td> 354</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>chr1</td><td>2368</td><td>17686</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td> 354</td><td> 354</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & gene\\_chrom & gene\\_start & gene\\_end & gene\\_id & gene\\_score & gene\\_strand & overlap\\_len & bidir\\_width & bidir\\_overlap\\_perc\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 & 2368 & 17686 & MIR6859-1:NR\\_106918.1 & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - &  410 &  410 & 1\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & chr1 & 2368 & 17686 & MIR6859-1:NR\\_106918.1 & . & - &  410 &  410 & 1\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - &  354 &  354 & 1\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & chr1 & 2368 & 17686 & MIR6859-1:NR\\_106918.1 & . & - &  354 &  354 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 15\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gene_chrom &lt;chr&gt; | gene_start &lt;int&gt; | gene_end &lt;int&gt; | gene_id &lt;chr&gt; | gene_score &lt;chr&gt; | gene_strand &lt;chr&gt; | overlap_len &lt;int&gt; | bidir_width &lt;dbl&gt; | bidir_overlap_perc &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 | 2368 | 17686 | MIR6859-1:NR_106918.1 | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - |  410 |  410 | 1 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | chr1 | 2368 | 17686 | MIR6859-1:NR_106918.1 | . | - |  410 |  410 | 1 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - |  354 |  354 | 1 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | chr1 | 2368 | 17686 | MIR6859-1:NR_106918.1 | . | - |  354 |  354 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand gene_chrom gene_start gene_end\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      chr1          0       29620   \n",
       "2 chr1  3917  4919 chr1:3917-4919 14    .      chr1       2368       17686   \n",
       "3 chr1  5632  6042 chr1:5632-6042 14    .      chr1          0       29620   \n",
       "4 chr1  5632  6042 chr1:5632-6042 14    .      chr1       2368       17686   \n",
       "5 chr1  6132  6486 chr1:6132-6486  7    .      chr1          0       29620   \n",
       "6 chr1  6132  6486 chr1:6132-6486  7    .      chr1       2368       17686   \n",
       "  gene_id               gene_score gene_strand overlap_len bidir_width\n",
       "1 WASH7P:NR_024540.1    .          -           1002        1002       \n",
       "2 MIR6859-1:NR_106918.1 .          -           1002        1002       \n",
       "3 WASH7P:NR_024540.1    .          -            410         410       \n",
       "4 MIR6859-1:NR_106918.1 .          -            410         410       \n",
       "5 WASH7P:NR_024540.1    .          -            354         354       \n",
       "6 MIR6859-1:NR_106918.1 .          -            354         354       \n",
       "  bidir_overlap_perc\n",
       "1 1                 \n",
       "2 1                 \n",
       "3 1                 \n",
       "4 1                 \n",
       "5 1                 \n",
       "6 1                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##load bidirectionals overlapping genes\n",
    "bidir_overlaps <- data.table::fread(\"/Users/rusi2317/projects/meta_analysis_qc_reviews/annotations/hg38_tfit_dreg_bidirectionals_in_genes.bed\",\n",
    "                                   sep=\"\\t\")\n",
    "colnames(bidir_overlaps) <- c('chrom','start','end','bidirs',\n",
    "                             'score','strand', 'gene_chrom',\n",
    "                             'gene_start','gene_end','gene_id',\n",
    "                             'gene_score','gene_strand','overlap_len')\n",
    "bidir_overlaps$bidirs <- paste0(bidir_overlaps$chrom,':',\n",
    "                              bidir_overlaps$start,'-',\n",
    "                              bidir_overlaps$end)\n",
    "\n",
    "##bidirectional regions widths that overlap with genes\n",
    "bidir_overlaps$bidir_width <- as.numeric(bidir_overlaps$end) - as.numeric(bidir_overlaps$start)\n",
    "bidir_overlaps$bidir_overlap_perc <- bidir_overlaps$overlap_len/bidir_overlaps$bidir_width\n",
    "\n",
    "nrow(bidir_overlaps)\n",
    "head(bidir_overlaps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3deWBU5bn48ScJQoCwKCkSCcim\nIAV3BSFFWSqyCOIGhgpalaLg+uu9KtbtSm1vtXVF21u1aq117WLrbnuxUu1yhSpYRdxwARGi\niKBsmfd3zplkkpkkJ0N4Mu/7ku/nj8xk5mwZzjeZSR4SMQB2mtg+AGBXQEiAAkICFBASoICQ\nAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQ\nAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQ\nAAWEBCggpJZlw+9W2D6EXZNfId0tIn9LXu0nUmbMT0WKbB3Mtqv67Nb+l9XvzY6OJyXjwJ4K\nDnyd9uE2ZWtL5brkFU+O1xu7YEiVEydO/EsODuam4GDkjur3vDgx7z92d2k95HubjCfH65Fd\nMKTtwUIP5OBgRovsMWdR9XsenJjbJkhS35VeHK9XPA8psX379syFchXSfiLn1byXcWJmHFjV\niVnf4Tbdjp6Y1wTlz5AxwSM30Yvj9YrnIdXjzUXBQtf87Yud2E1lVksNEPlezXuzGzqeSNWJ\nqWtHT8y+kvfqMrn+k/YiK304Xq94HlL1v03iN2P7FPYZc+82Y05MPn0JF1v//cM7t+lZ/kLV\n2psuHlw05t/BKkODd64S6Wd+c+C+xmy988i92/QYdlOY3gXBVl8Zltdm8K2JzVf1K+x7+qra\nu6+9vardXFV9X3hifnJGt8L9r9savps6aVacVNzu8N+kP1Wq2Xe6WtsPFmkTfS4YJDIluPjL\nSSVt+k79q6m9dvU+an8AZ4octfGC7m0Hzd9sMh6Yr/KkqwlCMt8UeczV462tzn31bdSY9747\nqku3o27YWmf9nNo1QqqcXPXs/+AvaoX08l5Vt86JvsS82z+83nVOTUj35Ukvs3lo1VKDv4hC\n6tE5eu/CUdFF9/U1e0/bXj0hDd43uunwDabmxPzfPaLbTq5zYkb7Tld7+8uDi98Gt30UXD5o\nzGVV9/xnotbaVVtL+wCCk294WfTe/sszHhjTUQreDUPatH79FkePN03mffVu1DzeLnlr/zVN\nOqW0+BdSjZqQbgze2/f4oXkip9W8RvqyT3Clx2FtgrffD97dflCYUftwxaqQ9uwiwb/EJeE/\nwpF7Bm+vjEISaZ88maRbQfDmv1I7T9/e64t6isxY9H71vbPDNfJ6haucZVIHtrZDuJ2iaHNp\nJ2Zy32nSt3+wyBnBjb8IDmeTeTq47YjTDwveXl9r7aqtpX0AwcmXFxxGq+C9MZkPTPCVqMsF\n4RbcPd40GffVv9GK4B9rt4MPDG49uqlnlYpdI6RxIt8y0Xt7JFIhfU8k/w5jVh0q0m5d9E8s\nNyS2nlsTkrQ+++f3h98zuDx4f7rIMcmQLq80VwcXPd8w7++VfJ6SlLG9uq+R5Otvm4rgbN1t\nZerALgzOq8fMtovrnJjJfadJ3/51IiXBZ95pIqeY7fsldxUsUVRRs3bV1tI+gDPDz9/vmU+P\nDi5fynhgXmkVfXa/83N3jzdN+n0NbDT49yxeaswDwfv/bvqJtfN2jZAOCZ6D3f6++XLhwoXb\nUiEFJ3p5uNLSvOhZR3DGDA/eqxxQE1L4XCTxwAMPrA0+FY8QOTQKqWsi+QTl5uDO02u/Is/Y\nXj0h/T24/CR4pvGD1IEFn+CvCO89qM6J+ds6H1z69j8I3v6fqSwW+X3wwkbyPwvu2RCU8Oua\ntZNbS/8AwpPv5eC+j9uKXJPxwJg/d48et7Y/qnT1eNOk39fARvcWuTS8PLpfv183ev40I/9C\nqnqNNKB2SJdHJ8i+Zz8S/qixKqQtwQP+SLToPiLXGtM5eht9RqsKqUNyS9sW/ddJfcP1kyEd\nEty0LnjvcZP+rbjM7dUNaa/oSvBF4NTqA/sw2M6/wht/lHlidqjzsWVuf0T4AuyfIp02m0dq\nffa4smbt6tc1tT+A4OTrEd05OjqMtAcmOAl/kXzlV+7q8aZJv6/+jX4V9PtknWOzYNcIactl\nX0s+wp1+ngrpndSThVHh0/fK4BG/O3zvf1Ih9Y3uXTIweCreb8ak6pCCt1FIT5n0kDK2V09I\nh0RXZomMqD6wPwXrrA1vfDjzxOxb52PL3P7tIoeZ+SKnG/PjWufQuTVrV20t7QMITr4h0Z0z\nohcNaQ9MaJmUh6/fn3P0eNOk31f/RpdXfdWybtcIKfgs9/wlB4QPcf6rtb8iPRrd1z/65lo3\nSb7OvqbWt78Dm4OXzNPeNObSxkLK3F5DX5EmisysPrC/B9t5NbzxzswTs1+djy1z+2tbSd7q\n4NP808Y8JNJ5UZW3a9ZObi39AwhOvr2jO4NnsudkPDChZXL91oHhp3RXj7eW9Pvq3+iG4ECf\nqXNsFuwSIX2xbNmy4KZVQSOyIPUaqX/yhbZ5LT96DnKkyNjw3aHpIf296pPwxMZCytxePa+R\n/hlcftpR5L+rD+zT4Marw3uPb/zErLP98SI37ibFwWubV4IONoZ3ffLxx5syT8z0DyD8Tlf4\n5Gxt+/BVXtoDc2NR0fjo50hzRc5083jTpd/XwEb3Sn7rwhw7aNAjdY8xd3aJkJZXfVpa00bk\niSik8N8k+JSXf48xqw8T2T34l/uBhM/tKq+U9JCeCd7/ozF/yGs0pIzt1RPSAR+YDcH50fqt\n1CkYvHwoesokovnWxk7MzO3/UqSLyNnBta29RS4KLh/Pk/x/Z56Y6R9A+AL9oI/M58Ern7Yf\npj8wwYuMDp+FIY0Ov4I4ebzp0u9rYKOnBXtfbMz9wbLvZHUONZNdIqTwp7MFI6YdEXzm6roh\n+rZCz7nvmE29gqX7DmsbvA1fIGwKnttJaSfJCGl1fnhG7R/8s8qg+JAytldPSJK/T2tJ/nNX\nHdhPw1tLd5dsTszM7X8RXpPnw6uPBlcOPP3Q/OhnPhknZvoHEP3spSA6jIszHpiNwWGMfkD+\n46LgvsWOHm+ajPvq3+iqDiK7HX5ocN9JjZw8zWvXCOn1LpJUGP47hj+WDxf7x55Vt56TCNdI\n/tC+0ynpIYU/Ogn0OVWk1dLYkDK3Vyekzm2ie08In4BU/xT/yOQax2ZxYmZu35wUXN0rOff3\nvYLkPadur3Nipn8Awck37PDohpM2ZT4wv86veke+7erxpsm8r96NmoeSH4UctqGeQ8ydXSMk\ns/6Gb/RpW3zwBR+F76ydWdJ2wGvBlU+/P7FXhyFnPF+19tun9+8y4ZXbM0KqvHn/9gdf9Plj\nwaYviA8pY3t1h1Zfnz6g8Os3RmdV6rsgj57cu3jc/fXMrtUj43h/Gx1S0l/LB7Xd54Tov1ll\nnphpH0A4n7ZhbkmbqsNIf2B+1zs65dpfvM3Z462tzn31bTR4Yn9uWeeSUT/PbtK42fgVkobL\nbT8JaFbhydfwvVv/fLFM+P2nuTucRsUdb/zH4piWE9Kcfv2GfGnMtoHJ0btdVGMn37LqWTtH\nEJJ3bg6eShz/9DNjgpdJ79o+luZDSJa0nJAqT6p6ady+7tiYHZdIbfW+CtlhjZ18ic1N/y+v\nzX68GTsgJEc9N2m/dnsOvehj28dR7f5ptZ2vss3bTzzxSpUN1aPZjzdjB835sajbyZAqVln+\nZgnghJ0IaemMvoUirUqnL2p8WWDX1vSQ5uZJyZDx44eWikyx/P/lAduaHNICGbs4eW3Z1OR/\n9AFariaHNKz/tuqribKYX+wEtARNDqnjzJrr8zopHAngsaZ/RRpQ8wOJkcNVjgXw1k68RhqX\n/D+XZvl0ma91OICfmv5du9kiPcomTR7RW2TSFsUjAjy0Ez9HWlLeq7VIQUn5Qr3DAfy0c5MN\niTWrmWwAGBECVDAiBChgRAhQwIgQoIARIUABI0KAAkaEAAWMCAEKGBECFDAiBChgRAhQwIgQ\noIARIUABI0KAAkaEAAWMCAEKGBECFDAiBChgRAhQwIgQoIARIUABI0KAAkaEAAWMCAEKGBEC\nFDAiBChonhGhLb/4WcqCy5p8cIAvmmdE6P39+qR0F37KhF1e848I/ZWQsOtr/hEhQkIL0Pwj\nQoSEFqD5R4QICS1A848IERJagOYfESIktADNPyJESGgBmn9EiJDQAjT/iBAhoQVo/t8iREho\nAZr/twgREloARoQABYwIAQoYEQIUMCIEKGBECFCwMyGteaPqO+BrP4xZipDQAjQ9pCX7i3S7\nJ7p6TNxWCAktQJNDeqswf8z4QrktvE5IcNCqZxuxSnFnTQ5pWt4TxnzSr/ANQ0hw0hmtd4/V\n+gzFnTU5pN7HhG+Xtz3WEBKcdNppO3f/DmlySG3OjC4ul+cJCU7yI6ReA6OLjT36rCckuMiP\nkM6TuRvCy8dl8meEBAf5EdK6fpI/OrxyuRQVExLc40dIpuKSAclnd3f3F0KCezwJqUbi3edi\n7iUk2OFdSPEICXYQEqCAkAAFhAQo8CKkWzqniVmSkGCHFyGtOK+NdBiUErMkIcEOL0Iy5imZ\nmNVyhAQ7PAnJ7EtIcJkvIU2fktVihAQ7fAkpS4QEOwgJUEBIgAJCAhQQEqCAkAAFhAQoICRA\nASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRA\ngU8hVayqbGwRQoIdvoS0dEbfQpFWpdMXxS5GSLDDk5Dm5knJkPHjh5aKTNkasxwhwQ4/Qlog\nYxcnry2bKtfGLEhIsMOPkIb131Z9NVFWFrMgIcEOP0LqOLPm+rxOMQsSEuzwI6RhA7anro8c\nHrMgIcEOP0JaIONeTV5bPl3mxyxISLDDj5DMbJEeZZMmj+gtMikuFUKCHZ6EZJaU92otUlBS\nvjB2MUKCHb6EFEisWc1kAxzlUUiMCMFdvoTEiBCc5klIjAjBbX6ExIgQHOdHSIwIwXF+hMSI\nEBznR0iMCMFxfoTEiBAc50dIjAjBcZ6ExIgQ3OZLSIYRIbjMo5AYEYK7fAmJESE4zZOQGBGC\n2/wIiREhOM6PkBgRguP8CIkRITjOj5AYEYLj/AiJESE4zo+QGBGC4zwJiREhuM2XkAwjQnCZ\nRyExIgR3+RISI0JwmichMSIEt/kREiNCcJwfIcWOCG26/ocpZxMSrPAjpNgRoVXHjEk5TDY3\ndR/ATvAjJEaE4Dg/QmJECI7zIyRGhOA4T0JiRAhu8yUkw4gQXOZRSIwIwV2+hMSIEJzmSUiM\nCMFtfoTEiBAc50dI/BYhOM6PkPgtQnCcHyExIgTH+RESI0JwnB8hMSIEx3kSEiNCcJsvIRlG\nhOAyj0JiRAju8iUkRoTgNE9CYkQIbvMjJEaE4Dg/QmJECI7zIyRGhOA4P0JiRAiO8yMkRoTg\nOD9CYkQIjvMkJEaE4DZfQjKMCMFlHoWUDUKCHYQEKCAkQAEhAQq8COmWzmliliQk2OFFSCvO\nayMdBqXELElIsMOLkIx5SiZmtRwhwQ5PQjL7EhJc5ktI06dktRghwQ5fQsoSIcEOQgIUEBKg\ngJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKg\ngJAABYQEKCAkQAEhAQoICVBASIACQgIUEBKggJAABYQEKPAppIpVlY0tQkiww5eQls7oWyjS\nqnT6otjFCAl2eBLS3DwpGTJ+/NBSkSlbY5YjJNjhR0gLZOzi5LVlU+XamAUJCXb4EdKw/tuq\nrybKymIWJCTY4UdIHWfWXJ/XKWZBQoIdfoQ0bMD21PWRw2MWJCTY4UdIC2Tcq8lry6fL/JgF\nCQl2+BGSmS3So2zS5BG9RSbFpUJIsMOTkMyS8l6tRQpKyhfGLkZIsMOXkAKJNauZbICjPAqJ\nESG4y5eQGBGC0zwJiREhuM2PkBgRguP8CIkRITjOj5AYEYLj/AiJESE4zo+QGBGC4/wIiREh\nOM6TkBgRgtt8CckwIgSXeRQSI0Jwly8hMSIEp3kSEiNCcJsfITEiBMf5ERIjQnCcHyExIgTH\n+RESI0JwnB8hMSIEx/kREiNCcJwnITEiBLf5EpJpeERo/bmzUiYTEqzwKKSGRoTWTj8pZZRs\n3rl9AE3iS0iMCMFpnoTEiBDc5kdIjAjBcX6ExIgQHOdHSIwIwXF+hMSIEBznR0iMCMFxfoTE\niBAc50lIjAjBbb6EZPgtQnCZRyHxW4TgLl9CYkQITvMkJEaE4DY/QmJECI7zIyRGhOA4P0Ji\nRAiO8yMkRoTgOD9CYkQIjvMjJEaE4DhPQmJECG7zJSTDiBBc5lFIgfX/XBu/ACHBDk9CWn/F\nxIs2mevbiRz+etxyhAQ7/Aipoq+ITPul7HPW+PzOa2IWJCTY4UdIF8pNH93TqvWEoJIn886O\nWZCQYIcfIQ0YGbyZKq+E10d/PWZBQoIdfoTU7jvBmyvky/D62e1jFiQk2OFHSANGBW+mSTTd\ncDRfkeAeP0K6UG775P5WrY7baswzed+JWZCQYIcfIVX0EZET75T9zj6uoOPqmAUJCXb4EZL5\n9NIJV2801xaKHLA0bjlCgh2ehFSl4sWP4hcgJNjhV0iNIiTYQUiAAkICFBASoICQAAWEBCgg\nJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCgg\nJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCiwFtLdnytuOIWQYIe1kKTw+Ie+VNx2EiHBDmsh\nLTgyX4q+9cetips3hARbLL5GWn1r0NIeZ/25MtvVK1Y1uighwQ6732xYfeuIfCk5/2+Nr7p0\nRt9CkVal0xfFLkZIsMPyd+3+dVVvCez7SCNrzs2TkiHjxw8tFZkS92yQkGCHxZC2/fn8vUVK\nZj/z8kVFeX+OXXGBjF2cvLZsqlwbsyAhwQ5rIT1y6u4ifb/7YiJ8Z7GcE7visP7bqq8myspi\nFiQk2GHv299ywFWvVL/zefF1sSt2nFlzfV6nmAUJCXZYC+n6t3dgxWEDtqeujxwesyAhwQ6L\nr5HefDZ489M3sllxgYx7NXlt+XSZH7MgIcEOeyGdnxe+2GmVd1EiizVni/QomzR5RG+RSXGp\nEBLssBbSXTLs8eDiiZFyZzarLinv1VqkoKR8YexihAQ7rIU0cp/kz4O2DTw0y7UTa1Yz2QBH\nWQup83eqrpzTIdvVGRGCs6yFNGBc1ZUJ+2azKiNCcJq1kGYV/C66fLIgmz0wIgS3WQupopeM\nmX/HD4/N67q68RUZEYLj7H37e+Wp+eG86oTXs1iRESE4zub09yeLfvXcB1mtyIgQHOfHLz9h\nRAiOsxfSI9PGVGl8RUaE4DhrId0hUlSclMWajAjBbdZC+nrH+J8IZWBECE6zFVKi9bk7ujYj\nQnCXrZA2y4U7vDojQnCWtad2R/ZevyOrMiIEp1kLaeXgwQ+9tS6SxZqMCMFt9qa/20u1xldk\nRAiOsxbSmTUaX5ERITjOj8kGRoTgOJshbXr1pSxXZEQIjrMX0nvH7xa8PLpi+kdZrMiIEBxn\nLaRVPWTYSDE353VflcWajAjBbdZCmiP3mvsk/B+y8b+tuAojQnCatZD2HmmikMykfbJcu6ER\nobXlJ6WMks07fZTAjrMWUvvvVIV0dvtsV29gRGj9+bNSJvMVCVZYC2nI4VUhHXxINqsyIgSn\nWQtpvlxTGYY0Xy7JYk1GhOA2ayFtHyH9jpBzDpHBXzW+IiNCcJy9nyNtubGniHT53oYsVmRE\nCI6zOiL0xWsV2a3IiBAc58esHSNCcJy1kL5Vo/EVGRGC4yz+DdkqHfplsSYjQnCbtZA2R9Y9\nN7zt49msyogQnGb9NdKmfbtsq+fmevBbhOAu6yGZ/5Dsfv83v0UIDrMf0vltGs3DMCIEx9kO\nKfGXTvtnsSYjQnCbtZCKktqI3N34iowIwXHWQppYZcZvs1iRESE4zvZTu+wwIgTH+RESI0Jw\nnLWQStPEPV0zjAjBedZCmt1dpOSQ0jzpVRaY0siajAjBbdZCeiH/6NeCizfGdn8vm1UZEYLT\nrIV0bO8vo8sv+5yY5dqMCMFd1kLac2bVlW+XZr+BNS9vjF+AkGCH1d9rFxldksWa78283Zh/\nHCiSf2zsZB4hwQ5rIU3LS/4k9vf5kxpfcUUXucW82Tbv6NkjpOSzmAUJCXZYC+m9Lvkn3/Xk\nXVPz277S+IontfqDMSfkPxtcfUDi/oozIcEOez+Q/deo6D/IDnouixX3PC54Uzouuj5mUMyC\nhAQ7bE42LHv4x/e+lM3/oTAdpgdvup4RXT+rY8yChAQ7/PhDYyO7f27MxP0TwdXKr4+IWZCQ\nYIcff2jsKRn6kllSNG+7+WoO/40CDvLkD43d2Ep6fqOPFB/aUSYyIgT3+PKHxlZe1KuNiOx+\nzNOxixES7PDoD42ZT99v9K+IERLs8OkPjWWBkGCHL39oLEuEBDs8+UNj2SIk2OHHHxrLGiHB\nDj/+0FjWCAl2+PGHxrJGSLDDVkif/fRFxQ2nEBLssBVSZfss/r7YjiMk2GHtqd25Xdcpbrka\nIcEOayElLhv80IoNG0OKeyAk2GEtpG7dCqr/+KXiHggJdlgL6bQainsgJNhhJaS59yhuNA0h\nwQ4rIUn0Hbu7zlTcdBVCgh0WQzpN88VRFUKCHYQEKCAkQAEhAQoICVBASIACOyHtPS3QW6Yl\nKe6BkGCHnZDSKe6BkGCHlZD+L53iHggJdlj9H7L6CAl2EBKggJAABYQEKCAkQAEhAQoICVBA\nSIACQgIUEBKgwKeQKlZVNrYIIcEOX0JaOqNvoUir0umLYhcjJNjhSUhz86RkyPjxQ0tFpmyN\nWY6QYIcfIS2QsYuT15ZNlWtjFiQk2OFHSMP6b6u+migri1mQkGCHHyF1nFlzfV6nmAUJCXb4\nEdKwAdtT10cOj1mQkGCHHyEtkHGvJq8tny7zYxYkJNjhR0hmtkiPskmTR/QWmRSXCiHBDk9C\nMkvKe7UWKSgpXxi7GCHBDl9CCiTWrGayAY7yKCRGhOAuX0JiRAhO8yQkRoTgNj9CYkQIjvMj\nJEaE4Dg/QmJECI7zIyRGhOA4P0JiRAiO8yMkRoTgOE9CYkQIbvMlJMOIEFzmUUiMCMFdvoTE\niBCc5klIjAjBbX6ExIgQHOdHSIwIwXF+hMSIEBznR0iMCMFxfoTEiBAc50dIjAjBcZ6ExIgQ\n3OZLSKbhEaFVZYek9JfNO7UPoGk8CqmhEaFNP/5hytl8RYIVvoTEiBCc5klIjAjBbX6ExIgQ\nHOdHSIwIwXF+hMSIEBznR0iMCMFxfoTEiBAc50dIjAjBcZ6ExIgQ3OZLSIbfIgSXeRQSv0UI\n7vIlJEaE4DRPQmJECG7zIyRGhOA4P0JiRAiO8yMkRoTgOD9CYkQIjvMjJEaE4Dg/QmJECI7z\nJCRGhOA2X0IyjAjBZR6FFLojfrCBkGCJZyHJ7Pj7CQl2eBHSB3+sJuOCNzFLEhLs8CKkuyVN\nzJKEBDu8CGnDaVI0L/pNqjIkeBOzJCHBDi9CMubhPXq/EG2B10hwkichmQ9G5V+6lZDgKl9C\nMonrWh+4jJDgKG9CMmbJfoU3EBLc5FFI5ss5Qkhwk08hGfPc9c/GL0BIsMOvkBpFSLCDkAAF\nhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAF\nhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqDAp5Aq\nVlU2tgghwQ5fQlo6o2+hSKvS6YtiFyMk2OFJSHPzpGTI+PFDS0WmbI1ZjpBghx8hLZCxi5PX\nlk2Va2MWJCTY4UdIw/pvq76aKCuLWZCQYIcfIXWcWXN9XqeYBQkJdvgR0rAB21PXRw6PWZCQ\nYIcfIS2Qca8mry2fLvNjFiQk2OFHSGa2SI+ySZNH9BaZFJcKIcEOT0IyS8p7tRYpKClfGLsY\nIcEOX0IKJNasZrIBjvIoJEaE4C5fQmJECE7zJCRGhOA2P0JiRAiO8yMkRoTgOD9CYkQIjvMj\nJEaE4Dg/QmJECI7zIyRGhOA4T0JiRAhu8yUkw4gQXOZRSIwIwV2+hMSIEJzmSUiMCMFtfoTE\niBAc50dIjAjBcX6ExIgQHOdHSIwIwXF+hMSIEBznR0iMCMFxnoTEiBDc5ktIpuERofcH9EnZ\nSzbv1D6ApvEopIZGhLbc/bOU/+QrEqzwJSRGhOA0T0JiRAhu8yMkRoTgOD9CYkQIjvMjJEaE\n4Dg/QmJECI7zIyRGhOA4P0JiRAiO8yQkRoTgNl9CMvwWIbjMo5D4LUJwly8hMSIEp3kSEiNC\ncJsfITEiBMf5ERIjQnCcHyExIgTH+RESI0JwnB8hMSIEx/kREiNCcJwnITEiBLf5EpJhRAgu\n8yakNW9UfQd87YcxSxES7PAkpCX7i3S7J7p6TNxWCAl2+BHSW4X5Y8YXym3hdUKCg/wIaVre\nE8Z80q/wDUNIcJIfIfU+Jny7vO2xhpDgJD9CanNmdHG5PE9IcJIfIfUaGF1s7NFnPSHBRX6E\ndJ7M3RBePi6TPyMkOMiPkNb1k/zR4ZXLpaiYkOAeP0IyFZcMSD67u7u/EBLc40lINRLvPhdz\nLyHBDu9CikdIsIOQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQ\nQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQ\nQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQ\nQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJECBTyFVrKps\nbBFCgh2+hLR0Rt9CkVal04a2yuYAAAwcSURBVBfFLkZIsMOTkObmScmQ8eOHlopM2RqzHCHB\nDj9CWiBjFyevLZsq18YsSEiww4+QhvXfVn01UVYWs2DDIb3zULwFt8bff9uCRta/Lf7+2xtZ\n//bb4+//2f3Pxrr/Z/Hr39bI9hs7/sY+/pt+8LNYl3/3h7HOPz/+/u+cfnGsk4+dFWvUQWNi\nDex9SKyee8ff32Vo/CnsRkgdZ9Zcn9cp4853vrZ7Sgdp6InfLAGaUf7usVqf0dSTvx5N/4o0\nYHvq+sjhGXdW/m/N5+Vn7mtoE6viP6E/++CD3M/9zXf/s6uaevLXYydeI417NXlt+XSZr3U4\ngJ+a/l272SI9yiZNHtFbZBLfTkALtxM/R1pS3qu1SEFJ+UK9wwH8tHOTDYk1qxudbABagOaf\ntQNaAEICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQ\nQEiAApshDbX965rQsjXye+92iM2Qyo/9P6uOZf8te//liiezzZBUf9Ml+2f/NvdPSOyf/Ssg\nJPbP/hUQEvtn/woIif2zfwWExP7ZvwJCYv/sXwEhsX/2r4CQ2D/7V0BI7J/9K7AZ0qxZFnfO\n/tm/5v5thvTppxZ3zv7Zv+b++W8UgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQ\nEqCAkAAFhAQoICRAASEBCgjJBV/c/YHtQ2hBVtzSDBvNfUi3De80/LbYG3K7/4oL+hT2OeUt\na/sPnSZ/tLf/p0d3LDnZ3sdfcdHAdgMvyuX/8Tu/c+wBNU3OQ5ot/WfsK3Njbsjt/iv6ycAz\nj85ru8TS/kMPS+5CqrP/m6SkfFJBl5WW9v9pHzlq1pHSb32O9m/MM23SQlI6/3Id0hI5ZpvZ\ndnTe0gZvyPH+L5U5wdvH8w+wtP/Ah3sU5SykOvtf2WpIcA4/JjMt7X+eLAje3ihX5mb/Znp/\nkdohaZ1/uQ6pXF4J3r5c8+9W54Yc739A/obwYoyssbN/YxKjes/LWUh19n+RvBRe3HK7pf1P\nkE+Ctx/JxNzs30yZOLFD7ZC0zr9ch9SrNLoo6dvgDTne/8DR0cV4ecPO/o25Lv+FH+YspLqP\nf88c7bmB/V8t9wdv75WLcncMg2qHpHX+5TikROvh0eWQwoZuyPH+q3xSuOc2S/tf0vpSk7OQ\n6u6/zTf+dWzXHieusLX/jSfsVn7lKa2mbcrNAYRqh6R2/uU4pDUyKbocLxUN3JDj/Sct7ye/\nyMXu69n/lwMP3JK7kOrs/1PpWzT428fkt/unnf0bc1crEWl9X052n1Q7JLXzL8chrZbJ0eV4\nWdXADTnef2jjFW0Lb83F3uvb/5zCZSZ3IdXZ//silyaMeTbvIDv7Nz+QSa9sWjJBfpyT/Udq\nh6R2/uU4pMqCEdHl0FaVDdyQ4/0HnugpE3PzAqme/T8nN5gchlRn/5uleHt4eXRuvtlSZ/8V\nhfttDS627NPu81zsP1I7JLXzL9ffbCjpE130KG3whhzv31whfZ7Pzb7r2//1qb9Vf4eV/Zvd\nD40uZsvLVvb/opwdXZ4puXluGUr7ZoPW+ZfrkE6R5cHbZVLe4A053v/dMjl3Pwusu/9nZ4eG\nyLjZi6zs3xzT8avwYkT+Riv7/0iOiy6T3wXPjbSQtM6/XIe0UL5lTGKqvGDM1nWfpd9gY/+J\n/kWf5WbP9e8/KXff/q6z/6dlbvCk5sFc/Rynzv4PKHgmePtk/mG52X+oKiTd8y/nI0ITZdS8\nEdGnoefkwPQbbOz/XSk+Jmmtlf0n5S6k+h7/wbO+KSW5mprN3P+rHfLGnj0mr9O/c7R/kwpJ\n9/zLeUibrz6i47D5W0zqA6m5wcb+/5R6jfKhlf0n5TCkuo///LIOA+fkbGi0zv5XnTWw3cDZ\nH+dq/yYzJKXzj/9GASggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiA\nAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiAAkICFBASoICQAAWEBCggJEABIQEKCAlQQEiA\nAkICFBASoICQAAWE5Ik7cvanwNAUhJSt4jGpq8fVPGplpWnv7rhvyeYslkq8Ir/6yvpRoGGE\nlC2bp/AD3USk8NoEITmLkLJV6xReV/P3ZsNTeN3O/PnZbE7hx2XMQnn0DLnN6lEgDiFlq9Yp\nXEt4Cler/GxHtxmsUfcUTmzNXGxE6VdGliUO3NfqUTR8axP2ucshpGwVj1l5cvfS418Prp4Y\nPmpvnlja/aR3w1M4fPe0bonzin6atsLn5+5fdOjFwSubclkUvv+gXGzM9muPKOp17mqTWiM6\nhVeeul9hjxP/Fe7lzLu6Ss+T3w5XuHdI5y5HPhVc6TYu+IdaZu75dsLCUdR6BGrdWmcLpmL2\nwM6j7tJ9zD1CSNkqHtCjz8xvSNELyXP2xY55R53ao9veqVP4sq+d8tfay3/cV4afdbAM+sL8\nTr4b3nBi0MKWI+XQWUdJz5WpNcJT+LWiNifMPrxgj4+CvfTJ6z1zRN7uLxvzfSk5ZUan/OeN\nGVKyMQzJylHUfgRqbq27hZW9CsbO6ifnN9vj7zhCylaxTAw+a98vhyXP2cPyf2PMhjKpPoUL\nBq9LX/4cuTF4e7FcYzZ37Bdc29TuEGNuDN415h45IbVGeAqfK48H1xbIveFexgVfPX4to4Or\n/YP9vSTfNuYnctjT1SHl+ijSHoHUrXW3cKoER7J1WN6b2o+7JwgpW8UF0XOa8fJKeM7+U04O\n3/tH6hSWB9MX39p6UCK42Nxtr/A0DSp4WG4ypme/yvDOI1pvql4jPIWfvy+89YnwnC/Oj87E\nCfLm1oLe4QuSZe8Gb/5focjgn3xs4SjSHoGaW+tsYW1+FN3jZc/u1IPsL0LKVnG/6OJmeSQ8\nZ++T5MuB4tQpvCJ98RUyN7o8XjaZx2S+MVNbrTEb5Yj7QiPl1eo1ql7mr376yr7RKdwrWusW\n+YM5TnrO+9Om5NY+vUHaSev/sXAUtR+B1K11t7Ao3HhLRkjZKi6LLn4jC8Jz9np5Mnr3oNQp\nvCF98YVVZ9YcWR48qzrUfFk00ZjXpNqL1WuEp/DH5V+ToiNPjk7hoam9fHl1H5F20z9Kbk8W\n39+1YGXuj6L2I5C6te4WfiV37MyD6z9CylbxPtHFguCVRHDOPii/iN7tlTqFN6Yv/qacG12e\nGJ5np8r7j8pDxqyTOakFqtYIT+GheXP+Umn+Fp3CvaM7b5NHw4vldx4pgxOb31gffdfuIbnP\nylGkHoHUrXW38Jz8d5aP4y6KkLJVXPBueDFJ3g7P2cUyLXzvnfyGTuGtrQ4IL7Z07xa8/YPc\nekqncMSny6HRnTdfUfsUfj98zW7MM8lXJ2+F14+TpSu+90x47Zuy8m35URTSYrku50eR9gjU\n3FpnC+/LpPCGJ1ulf/O95SCkbBXL5C3hj2EmJM/ZofmPGfPVBGnoFDazoudFl8nVwdstnYZ3\nmGWid78fvH06r7z2KVwRfXesYoT8JNzLhODVyiN5w8zbcsDWoIRD22z5arfJUUjXyzM5P4r0\nRyB1a90tjM8LnmVuG5X3ht4j7hVCylZxV9nnjFF5XV+r/glO/jfP6FtU1uApvKqXjDxniOwf\n3T5Tkj8O3TBIDjlnckH3D9OeVI2VCdfMKh4t+//RFJfu0e+MMXmd/h5+1ek36+x95HJjzpNL\n1smSh9sP3p77o6j9CNTcWncLr3ctmDhnoFyo/Kh7g5CyVTz3qWOKe03/wFTNFCw/oUe34xfP\nbvAUNuvnDG5/8CXRzLZ5XPokb/zy4oPb9Z29yqSdwhWzSzt+4x5zTqczTfFRKyZ3635C+I2w\nDVfu1754+K8SxnwxUwpkNznobQtHUfsRqHVrnS2YVTP2KTr454mdeIS9RkhuKT6q3ptX/FKu\n+Gu99+TwKBo4NoQIyS0NnqxVkw02j4KQYhCSWxo8WS/52PpREFIMQlJ0vdT4btM2MfB4p44i\nbVsKx7brIiRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCA\nkAAFhAQoICRAASEBCggJUEBIgAJCAhQQEqCAkAAFhAQoICRAASEBCggJUPD/AdActF6gDWDY\nAAAAAElFTkSuQmCC",
      "text/plain": [
       "Plot with title “Histogram of bidir_overlaps$bidir_overlap_perc”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "hist(bidir_overlaps$bidir_overlap_perc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Number of bidirectionals overlapping genes (250 upstram TSS AND 15kb downstream TES) = 589176\"\n"
     ]
    }
   ],
   "source": [
    "#the number of bidirectional transcripts overlapping genes\n",
    "print(paste(\"Number of bidirectionals overlapping genes (250 upstram TSS AND 15kb downstream TES) =\", \n",
    "            nrow(unique(bidir_overlaps[,c(1,2,3,4)]))))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "847521"
      ],
      "text/latex": [
       "847521"
      ],
      "text/markdown": [
       "847521"
      ],
      "text/plain": [
       "[1] 847521"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##check the total bidirectionals\n",
    "nrow(bidir_no_overlaps) + nrow(unique(bidir_overlaps[,c(1,2,3,4)]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gene to gene distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "95357"
      ],
      "text/latex": [
       "95357"
      ],
      "text/markdown": [
       "95357"
      ],
      "text/plain": [
       "[1] 95357"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>chr1</td><td>14361</td><td>29370</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>     0</td></tr>\n",
       "\t<tr><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>  2960</td></tr>\n",
       "\t<tr><td>chr1</td><td>14361</td><td>29370</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>     0</td></tr>\n",
       "\t<tr><td>chr1</td><td>14361</td><td>29370</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>  -996</td></tr>\n",
       "\t<tr><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>  2960</td></tr>\n",
       "\t<tr><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>-12930</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + & chr1 & 14361 & 29370 & WASH7P:NR\\_024540.1    & . & - &      0\\\\\n",
       "\t chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - &   2960\\\\\n",
       "\t chr1 & 14361 & 29370 & WASH7P:NR\\_024540.1    & . & - & chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + &      0\\\\\n",
       "\t chr1 & 14361 & 29370 & WASH7P:NR\\_024540.1    & . & - & chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + &   -996\\\\\n",
       "\t chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + &   2960\\\\\n",
       "\t chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + & -12930\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 13\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | V11 &lt;chr&gt; | V12 &lt;chr&gt; | V13 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + | chr1 | 14361 | 29370 | WASH7P:NR_024540.1    | . | - |      0 |\n",
       "| chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - |   2960 |\n",
       "| chr1 | 14361 | 29370 | WASH7P:NR_024540.1    | . | - | chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + |      0 |\n",
       "| chr1 | 14361 | 29370 | WASH7P:NR_024540.1    | . | - | chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + |   -996 |\n",
       "| chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + |   2960 |\n",
       "| chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + | -12930 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2    V3    V4                    V5 V6 V7   V8    V9   \n",
       "1 chr1 11873 14409 DDX11L1:NR_046018.2   .  +  chr1 14361 29370\n",
       "2 chr1 11873 14409 DDX11L1:NR_046018.2   .  +  chr1 17368 17436\n",
       "3 chr1 14361 29370 WASH7P:NR_024540.1    .  -  chr1 11873 14409\n",
       "4 chr1 14361 29370 WASH7P:NR_024540.1    .  -  chr1 30365 30503\n",
       "5 chr1 17368 17436 MIR6859-1:NR_106918.1 .  -  chr1 11873 14409\n",
       "6 chr1 17368 17436 MIR6859-1:NR_106918.1 .  -  chr1 30365 30503\n",
       "  V10                   V11 V12 V13   \n",
       "1 WASH7P:NR_024540.1    .   -        0\n",
       "2 MIR6859-1:NR_106918.1 .   -     2960\n",
       "3 DDX11L1:NR_046018.2   .   +        0\n",
       "4 MIR1302-2:NR_036051.1 .   +     -996\n",
       "5 DDX11L1:NR_046018.2   .   +     2960\n",
       "6 MIR1302-2:NR_036051.1 .   +   -12930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#load the gene to gene distances\n",
    "gene_dist <- data.table::fread('/Users/rusi2317/projects/meta_analysis_qc_reviews/annotations/hg38_refseq_diff53prime_genes_distances.bed')\n",
    "nrow(gene_dist)\n",
    "head(gene_dist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "10709"
      ],
      "text/latex": [
       "10709"
      ],
      "text/markdown": [
       "10709"
      ],
      "text/plain": [
       "[1] 10709"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'DDX11L1:NR_046018.2'</li><li>'WASH7P:NR_024540.1'</li><li>'LOC107985728:NR_168405.1'</li><li>'SAMD11:NM_001385640.1,NM_001385641.1'</li><li>'SAMD11:NM_152486.4'</li><li>'NOC2L:NM_015658.4'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'DDX11L1:NR\\_046018.2'\n",
       "\\item 'WASH7P:NR\\_024540.1'\n",
       "\\item 'LOC107985728:NR\\_168405.1'\n",
       "\\item 'SAMD11:NM\\_001385640.1,NM\\_001385641.1'\n",
       "\\item 'SAMD11:NM\\_152486.4'\n",
       "\\item 'NOC2L:NM\\_015658.4'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'DDX11L1:NR_046018.2'\n",
       "2. 'WASH7P:NR_024540.1'\n",
       "3. 'LOC107985728:NR_168405.1'\n",
       "4. 'SAMD11:NM_001385640.1,NM_001385641.1'\n",
       "5. 'SAMD11:NM_152486.4'\n",
       "6. 'NOC2L:NM_015658.4'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"DDX11L1:NR_046018.2\"                 \n",
       "[2] \"WASH7P:NR_024540.1\"                  \n",
       "[3] \"LOC107985728:NR_168405.1\"            \n",
       "[4] \"SAMD11:NM_001385640.1,NM_001385641.1\"\n",
       "[5] \"SAMD11:NM_152486.4\"                  \n",
       "[6] \"NOC2L:NM_015658.4\"                   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genes_overlap_opposite_strands_dt <- subset(gene_dist, V13==0)\n",
    "genes_overlap_opposite_strands <- unique(c(genes_overlap_opposite_strands_dt$V4,\n",
    "                                                  genes_overlap_opposite_strands_dt$V10))\n",
    "length(genes_overlap_opposite_strands)\n",
    "head(genes_overlap_opposite_strands)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "78579"
      ],
      "text/latex": [
       "78579"
      ],
      "text/markdown": [
       "78579"
      ],
      "text/plain": [
       "[1] 78579"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>  2960</td></tr>\n",
       "\t<tr><td>chr1</td><td>14361</td><td>29370</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>  -996</td></tr>\n",
       "\t<tr><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>chr1</td><td>11873</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>  2960</td></tr>\n",
       "\t<tr><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>-12930</td></tr>\n",
       "\t<tr><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>chr1</td><td>14361</td><td>29370</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>  -996</td></tr>\n",
       "\t<tr><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>chr1</td><td>34610</td><td>36081</td><td>FAM138A:NR_026818.1  </td><td>.</td><td>-</td><td>  4108</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - &   2960\\\\\n",
       "\t chr1 & 14361 & 29370 & WASH7P:NR\\_024540.1    & . & - & chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + &   -996\\\\\n",
       "\t chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & chr1 & 11873 & 14409 & DDX11L1:NR\\_046018.2   & . & + &   2960\\\\\n",
       "\t chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + & -12930\\\\\n",
       "\t chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + & chr1 & 14361 & 29370 & WASH7P:NR\\_024540.1    & . & - &   -996\\\\\n",
       "\t chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + & chr1 & 34610 & 36081 & FAM138A:NR\\_026818.1   & . & - &   4108\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 13\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | V11 &lt;chr&gt; | V12 &lt;chr&gt; | V13 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - |   2960 |\n",
       "| chr1 | 14361 | 29370 | WASH7P:NR_024540.1    | . | - | chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + |   -996 |\n",
       "| chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | chr1 | 11873 | 14409 | DDX11L1:NR_046018.2   | . | + |   2960 |\n",
       "| chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + | -12930 |\n",
       "| chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + | chr1 | 14361 | 29370 | WASH7P:NR_024540.1    | . | - |   -996 |\n",
       "| chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + | chr1 | 34610 | 36081 | FAM138A:NR_026818.1   | . | - |   4108 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2    V3    V4                    V5 V6 V7   V8    V9   \n",
       "1 chr1 11873 14409 DDX11L1:NR_046018.2   .  +  chr1 17368 17436\n",
       "2 chr1 14361 29370 WASH7P:NR_024540.1    .  -  chr1 30365 30503\n",
       "3 chr1 17368 17436 MIR6859-1:NR_106918.1 .  -  chr1 11873 14409\n",
       "4 chr1 17368 17436 MIR6859-1:NR_106918.1 .  -  chr1 30365 30503\n",
       "5 chr1 30365 30503 MIR1302-2:NR_036051.1 .  +  chr1 14361 29370\n",
       "6 chr1 30365 30503 MIR1302-2:NR_036051.1 .  +  chr1 34610 36081\n",
       "  V10                   V11 V12 V13   \n",
       "1 MIR6859-1:NR_106918.1 .   -     2960\n",
       "2 MIR1302-2:NR_036051.1 .   +     -996\n",
       "3 DDX11L1:NR_046018.2   .   +     2960\n",
       "4 MIR1302-2:NR_036051.1 .   +   -12930\n",
       "5 WASH7P:NR_024540.1    .   -     -996\n",
       "6 FAM138A:NR_026818.1   .   -     4108"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#remove gene overlapping other opposite strand genes\n",
    "gene_dist_other_overlaps <- subset(gene_dist, V13!=0)\n",
    "nrow(gene_dist_other_overlaps)\n",
    "head(gene_dist_other_overlaps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3deYBT5bn48WcGhAEGEKHi1AHZ\nFKSi1o1tiqK0CiKIK0JFXK5FxaW2datarWht9VZrtbW9arFa6271p3WBWm1Re7uAAl53rRu4\nAFI2EZg5v3OSzMxJGE6G5H0nzwPfzx/JmeTNOScxX5KceZ1IAKBoUuodALYEhAQ4QEiAA4QE\nOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIAD\nhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBI\ngAOEBDhASIADhAQ4QEhorhV/eKPUu6AXISWYKSJ/Sy/2E6kJgptFKku1M+sv67NNhzu8rPqJ\n8H4uyX/vFsg1qfMnw+H3py/6c7j4+8zVGzZ42TkjCClBs0KqHTt27F9aYGd+Fu6M3OJl1Qkh\nNd67uw7rIm0GX7w6THo7kZPSV18gUrEyvfgLkeXppQfGVLfb9agXvOyqWoSUoFkhbQgH3d0C\nO3OQyHZnzPGy6oSQ6u/d+kMlre+7QXCiSFX66j1FjkgvvbFjJqTab2ZGXuhlX7UipAQbhVS3\nYeP3Ly0V0q4iZ3ladSakpHt3RZjxFBkVPgxjg+Cx8MJ50bWLy9Lv7FYsuLyrZEL6ebjwpeGd\nwtPHPO2uSoSUYKOQmvD6nHDQFX9bWcRmaps1aoDIxUVsJEkmpCY03Lu+UjZ/oVz7SQeRd4Mv\nOotcFV09M/3O7vbMa1AqpPCROiUIPqluePu3dSCkBJt8a1f34MF9KvqM+u36IDgq/RyKhi2/\ncr9t2/ac9NfMrVefP6hy1P+FNxkS/nCZSL/gwT13CYJ1t+6/U9sew34WpXdOuNaXhpW1HXRj\n3drL+lX0PXFRfPPx9WU2c1njtVmrD4K/HF3Vtu+xzwWZbdVeO7Ci15GvBrnX5Xjj6G7t93sw\n561dE/fu8zLZPghDCr4u8kgQHJ/5Z2Vi+p3dzFhIa8Lz+eH5aSKDi3z4TSGkBJsKqXZ85pmz\n18pYSP/6cubSM1IvMe/0j5a3P6MxpDvLpFewdkhm1KCVqZB6bJv66dsHps52XN649az1bRRS\n9uqD72fGnleX2lbf41I/tX0p97psf94udcUxWSE1ee86Sat3opBWL1/+RRA8LNLqs3Dkdul3\ndotmz549IxPS6ssuu+zz8PxokYle/qMoRUgJ6v+lTWsM6frwp12OGBJ+QJja+CliTZ9woce+\nbcPTK8MfN3w1ep53iG6YCal7+DGiV3ScS/rv3z08/UEqJJEO6Wez7NAqPPlhw8az1/fKnJ4i\nU+a8V39tzuqjI9JDT9w3PL02tS2RsqrW4enBuddl+bRjtNnK1NaXJN+78JWo6zkNK1gb3vCe\nIPhb4zG74FFpOGoXqrurtbSf5/a/hm6ElGBTIY0W+WaQ+mm7uoan2sUi5beE/zrvI9I+fFb+\nJrz4urp1ZzaGJG1O+5+7omMGl4Q/TxY5JB3SJbXB5eFZz1eD98KXoAkNG89ZX85npOzVb9g1\nfWV4m8qlqW1N+DRYErbTKfe6LN8OK34kWH9+dkhN3ruXoixl91v/k77lcanKLms4ZpcT0uFh\n4VVb1/FvQkqwqZD2Dt+D/fK9YM0zzzyzvuGpFj7RJ0U3WhD+U/5Q6p/w4eFPtQMaQwovDeru\nvvvuT8PXghEi+6RC2j58v/VheOUNQeqwcuMRjZz15YSUvfqFYXThW61gRevovVa4rTbRC8Ud\nqT6yr8sSvq5cGp1/NSukJu9d8PSOqQeh3U9S71sfCF/J6oIhsTVmhbRP9JI4dY2T/whGEFKC\nmdLwGWlAPKRLUs+pXU67f3XQ8Obni9b1v+7fOXVIa9vMga2LG0LqmF7T+jk/PLpvdPt0SHuH\nFy2R9LHiabGQcteXE1L26u+P9f6DaFt9outmhT99lHNd3AfhJS9GCz/JCqmpexda8Zv0x7hU\n3WvCl5x/LS1vfGeXHdLzvzu3jcjhBT/wBhFSgk2F9MX3v5R+Ynb+n4an2tvhWfrNTPh8Ozmo\nDV9HZkY//bohpL6pa+cNDP+x7jdlXH1I4WkqpCeC7JBy1pcTUs7q/zsWy5npI4Sh2amQsq+L\n+1N4yafRwn1ZITV171IWyqSa8MfZ0fLRIjPujr2zy/mMlD4k/nKhD7xBhJRgUyGFLyvPXrBH\n9Fwrnx9/RXogdV3/1MG1HTKf7a+IHf4Ore0jMvH1ILgwX0i568t5Rcpe/b0i287JeCsnpOzr\n4v5X0keqg1sle2bDxvcuZaFcu25g5mXtHpFhU+PvFetD+uj++1O7HR0Gv6/gR94eQkqwiZBW\nLly4MLxoUfgklpsanmr905/Rg5fLU+/J9k8fMos+R8RD+t/Mq8DYfCHlri8npOzVvxQ+6VdF\nP37y0Uerc0LKvi5uWXj95dHCEfGQmrx311dWjkn9Hml66tetQbCqnbTqGntn1xDSK+H52+H5\nq+H580U+/pYQUoJNhPRaePFT4WUftxX5Y+qpFh0pCF9jym8PgsX7inQJU/mRRG++an8g2SE9\nFf78aBD8v7K8IeWsLyek7NWv6y1ybnjpY2VS/n85IWVflyX89FX5RFCXmg67JPHehR+0On4W\nhXRQ/e+yJkQ3anxn1xBSbaXI6KXBhyNEWmd3u2UjpASbemvXT6TViIlDww/c269Ife7vOf3t\nYHWvcHTfYe3C0/CzRbA6fPMl1Z0lJ6TF4euL7LF72JHslhxSzvpyQspZ/QPh+Z4n7hOu/L+C\nnJCyr8tyc3Tz6i6SFVKT925VOOigu+V754Yj56Zu+7voRrGjgA2fkaIjF+XbR3fwUlf/HSwg\npASbCumVrpkP7xXPhj9F8wKiYX/vnrn09NQMgvSsgc7HZYcU/e4m1Of48B/sBYkh5a4vZ65d\n9uqDi1ulxx6/YaOQsq7Lsnb/9BWHZYXU5L37fXnmwvoZdP9pK/F3do0h1R6TGXjU+iIfflMI\nKcGmQgqWX/e1Pu267XXOh9EPn55Q1W5AdIBq2ZVje3UcfPKzmVu/dWL/roe+9MuckGpv2L3D\nXuf+55Fw1eckh5SzvtxJq1mrD4LnJu3WbucjU//rUG5I8etyPHBM726j78qZa9fkvftD71Qd\nHc6vz+Ow+G+Ps47aPTZu53YDj5jVzAd5C0FInl0icrTd1cese/p8OfThZS2zMXsIyZMz+vUb\nvCYI1g9MT70ztvomLdxosh4aEJInN0QHtZ58alT4OeYde6tvEiElICRPao/OfObu8JCe1V8g\ncf02c5t1a7fqP2+SjJC8mT1u1/bdh5z7kaLV3zUx7mxPe7ZVIiTAAUICHCAkwAFCAhwgJMAB\nQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAk\nwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIc\nICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFCAhwgJMABQgIcICTAAUICHCAkwAFC\nAhwgJMABQgIcICTAAUICHCAkwAFCsmvRrPyerSv1Xm4lCMmuk9t0yWdbebnUe7mVKDKkpYtq\n3ewHNt/UqXmHLJWXWmBHUFRIC6b0rRBpXT15jrvdwWYgJEUKD2l6mVQNHjNmSLXIhHUO9wjN\nRUiKFBzSTXLw3PTSwmPlKle7g81ASIoUHNKw/uvrF+tqatzsDDYLISlScEidTmhcvqizgz3B\n5iIkRQp/RRqwoWF55HAn+4LNQ0iKFPEZafT89NJrk2WGq93BZiAkRQo/ajdNpEfNuPEjeouM\n+8LhHqG5CEmRIn6PNG9SrzYiraomPeNud7AZCEmR4mY21H28mJkNJUNIijBFyC5CUoQpQnYR\nkiJMEbKLkBRhipBdhKQIU4TsIiRFmCJkFyEpwhQhuwhJEaYI2UVIijBFyC5CUoQpQnYRkiJM\nEbKLkBRhipBdhKQIU4TsIiRFmCJkFyEpwhQhuwhJEaYI2UVIiviZIvTh0L0b7N6d4xF+EJIi\nfqYIfX7d1Q1OE35d6wchKeJ/itBzhOQJISnif4oQIflCSIr4nyJESL4QkiL+pwgRki+EpIj/\nKUKE5AshKeJ/ihAh+UJIivifIkRIvhCSIv6nCBGSL4SkiP8pQoTkCyEp4v+vCBGSL4SkiP+/\nIkRIvhCSIkwRsouQFGGKkF2EpAhThOwiJEWYImQXISlS5BShoPb1l9cnjyAkXwhJkYJDuvjW\n8GT91ZUibb+1PGkgIflCSIoUHJIcEJ5Mly5HfWuIDFybMJCQfCEkRYoKaWHZfkvCxVvl0oSB\nhOQLISlSVEi/ludSy8P2SxhISL4QkiJFhXSprEwtT+uYMJCQfCEkRYoK6Q5ZmFo+fFDCQELy\nhZAUKTykL19x39+/NDFa/Ps2JyUMJCRfCEmRgkPqUSaRp4PgwnZt5ycMJCRfCEmRwn8hu2b+\n/T86qeYvQTCgx1NJ4wjJF0JSpNiZDaGXk2cJEZIvhKSIg5DyICRfCEkRQrKLkBQhJLsISRFC\nsouQFCEkuwhJEUKyi5AUISS7CEkRQrKLkBQhJLsISRFCsouQFCEkuwhJEUKyi5AUISS7CEkR\nQrKLkBQhJLsISRFCsouQFCEkuwhJEUKyi5AUISS7CEkRQrKLkBQhJLsISRFCsouQFCEkuwhJ\nEUKyi5AUISS7CEkRQrKLkBQhJLsISRFCsouQFCEkuwhJEUKyi5AUISS7CEkRQrKLkBQhJLsI\nSRFCsouQFCEkuwhJEUKyi5AUISS7CEkRQrKLkBQhJLsISRFCsouQFCEkuwhJEUKyi5AUISS7\nCEkRQrKLkBQhJLsISRFCsouQFCEkuwhJEUKyi5AUISS7CEkRQrKLkBQhJLsISRFCsouQFCEk\nuwhJEUKyi5AUKTKkpYtq8w0hJF8ISZEiQlowpW+FSOvqyXMShxGSL4SkSOEhTS+TqsFjxgyp\nFpmwLmEcIflCSIoUHNJNcvDc9NLCY+WqhIGE5AshKVJwSMP6r69frKupSRhISL4QkiIFh9Tp\nhMblizonDCQkXwhJkcJfkQZsaFgeOTxhICH5QkiKFPEZafT89NJrk2VGwkBC8oWQFCn8qN00\nkR4148aP6C0yLikVQvKFkBQp4vdI8yb1aiPSqmrSM4nDCMkXQlKkuJkNdR8vZmZDyRCSIkwR\nsouQFGGKkF2EpAhThOwiJEWYImQXISnCFCG7CEkRpgjZRUiKMEXILkJShClCdhGSIkwRsouQ\nFGGKkF2EpAhThOwiJEWYImQXISnCFCG7mhHSEjnp/Lweb4F93eIxRciuZoT0puw+Kp/eo1pg\nX7d4TBGyq1kh3Zd3zMWE5ABThOwiJEX8TBF6f7+9G/SXtYVuA4kISRE/U4TW3vqrBufxiuQJ\nISnCFCG7CEkRpgjZRUiKMEXILkJShClCdhGSIkwRsouQFGGKkF2EpAhThOwiJEWYImQXISnC\nFCG7CEkR/oqQXYSkCH9FyC5CUoQpQnYRkiJMEbKLkBRhipBdhKQIU4TsIiRFipwiFLoleWID\nIXlDSIoUH5JMS76ekHwhJEUKDen9R+vJ6PAkYSQh+UJIihQa0kzJkjCSkHwhJEUKDWnFVKm8\n6OqIDA5PEkYSki+EpEjhn5Hu2673X1Nr4DNSiRCSIkUcbHj/wPIL1xFS6RCSIsUctau7ps2e\nCwmpZAhJkeIOf8/bteI6QioVQlKkyN8jrTlDCKlUCEmRon8hO/vaWckDCMkXQlKk+JkN+RCS\nL4SkCCHZRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVIihCSXYSkCCHZRUiKEJJdhKQIIdlFSIoQ\nkl2EpAgh2UVIihCSXYSkCCHZRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVIihCSXYSkCCHZRUiK\nEJJdhKQIIdlFSIoQkl2EpAgh2UVIihCSXYSkCCHZRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVI\nihCSXYSkCCHZRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVIihCSXYSkCCHZRUiKEJJdhKQIIdlF\nSIoQkl2EpAgh2UVIihCSXYSkCCHZRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVIihCSXYSkCCHZ\nRUiKEJJdhKQIIdlFSIoQkl2EpAgh2UVIihQZ0tJFtfmGEJIvhKRIESEtmNK3QqR19eQ5icMI\nyRdCUqTwkKaXSdXgMWOGVItMWJcwjpB8ISRFCg7pJjl4bnpp4bFyVcJAQvKFkBQpOKRh/dfX\nL9bV1CQMJCRfCEmRgkPqdELj8kWdEwYSki+EpEjhr0gDNjQsjxyeMJCQfCEkRYr4jDR6fnrp\ntckyI2EgIflCSIoUftRumkiPmnHjR/QWGZeUCiH5QkiKFPF7pHmTerURaVU16ZnEYYTkCyEp\nUtzMhrqPFzOzoWQISRGmCNlFSIowRcguQlKEKUJ2EZIiTBGyi5AUYYqQXYSkCFOE7CIkRZgi\nZBchKcIUIbsISRGmCNlFSIowRcguQlKEKUJ2EZIi8ZBm/mezb84UoRIiJEXiIUnFEfeuaf5N\nmSJUYoSkSDykm/Yvl8pvPpo03yeGKUKlRkiKZH9GWnxj2NJ2//V03vdrTBFSgJAU2ehgw+Ib\nR5RL1dl/y3dDpgiVHCEpsvFRuxcv6y2hXe5PvmHiFKH3du3T4Muytui9RFMISZHskNY/ffZO\nIlXTnvrXuZVlTyfeMHGK0Bczf9XgPF6RPCEkReIh3X98F5G+332+LvphrpyeeEOmCJUcISmS\ndfhb9rjspfof/tPtmuRbMkWo1AhJkXhI1761WTdlilCJEZIi2Z+RXp8Vntz8arNvzRShUiIk\nRbJCOrssOo7duuzcuubenClCJURIisRDuk2GPRae/XGk3NqcmzJFqMQISZF4SCN3Tk/1WT9w\nn2bckilCpUZIisRD2vZbmYXTO+a/IVOESo6QFImHNGB0ZuHQXfLfkClCJUdIisRDOrXVH1Ln\nj7fK/1+IvyJUeoSkSDykpb1k1Ixbrj6sbPvF+W/IXxEqOUJSJOvw97vHl0fzVQ99pRk3ZIpQ\nyRGSIjmzvz+Z87vZ7zfvlkwRKjVCUoS/ImQXISmSFdL9E0dlNPPWTBEqJUJSJB7SLSKV3dI2\nYw3L8qRESL4QkiLxkL7SKXmyT7bPfz51+oLgoS9L5fgPk8YRki+EpEgspLo2Z27GDT8bKOEL\n2Oy2nUbuJt2XJQwkJF8ISZFYSGvl25txw+/KufNn9WvXM3w1+r18J2EgIflCSIrE39rt33t5\n8284cGh48qD8KFo+YM+EgYTkCyEpEg/p3UGD7n1zSUr+G7Y7LTyZL/dGy6e1TxhISL4QkiJZ\ns787SL38N+x9UHiyctqL0fKRSUf5CMkXQlIknswpjfLf8JhtHqlffLPdmISBhOQLISlS8MyG\nt9pLn4ejhQVndS77c8JAQvKFkBTJCWn1/Beae8s3juh+Y3R+s3S/N2kcIflCSIpkhfTvI7YJ\nPx5dOjnxF6wxqTkNbz6X/PUVhOQLISkSD2lRDxk2UoIbynZc5HALhOQLISkSD+kM+W1wp0T/\nh2zyXyvePITkCyEpEg9pp5FBKqRg3M4Ot0BIvhCSIvGQOnwrE9JpHRxugZB8ISRF4iEN3i8T\n0l57O9wCIflCSIrEQ5ohV9RGIc2QCxxugZB8ISRF4iFtGCH9hsrpe8ugzx1ugZB8ISRFsn6P\n9MX1PUWk68UrXG6BkHwhJEVypwitfHmp4y0Qki+EpEgRf0WomQjJF0JSJB7SNxs53AIh+UJI\nimR/h2xGx34Ot0BIvhCSIvGQ1qYsmT283WMOt0BIvhCSIk19Rlq9S9f1TVxcIELyhZAUafJg\nw/ekmX//uzkIyRdCUqTJkM5um/cPETcfIflCSIo0EVLdXzrv7nALhOQLISkSD6kyra3ITIdb\nICRfCEmReEhjM6Y85HILhOQLISnCzAa7CEkRQrKLkBSJh1SdpcbRFgjJF0JSJB7StB1Fqvau\nLpNeNaEJjrZASL4QkiLxkP5a/o2Xw7NXD97x3w63QEi+EJIi8ZAO670mdb6mz1EOt0BIvhCS\nIvGQup+QWTip2uEWCMkXQlIk9+/apRxU5XALhOQLISkSD2liWfo3sQ+Xj3O4BULyhZAUiYf0\n767lx9z2+G3Hlrd7yeEWCMkXQlIk6xeyLx6Y+h9kd5vtcguE5AshKZIzs2Hhff/92xcc/j8U\nASH5Q0iKFP5FY81FSL4QkiJFfdFYsxCSL4SkCF80ZhchKcIXjdlFSIrwRWN2EZIifNGYXYSk\nCF80ZhchKcIXjdlFSIrwRWN2EZIifNGYXYSkCF80ZhchKRIL6bObn/exBULyhZAUiYVU28Hl\n94s1ICRfCEmR+Fu7M7df4mELhOQLISkSD6nu+4PufWPFqojDLRCSL4SkSDykHXZoVf/llw63\nQEi+EJIi8WSmNnK4BULyhZAUqQ9p+u2+tkBIvhCSIvUhSeqI3W2nuN8CIflCSIpkhzTVw5dT\nEJIvhKQIIdlFSIoQkl2EpAgh2UVIihQZ0tJFef8IHiH5QkiKFBHSgil9K0RaV0+ekziMkHwh\nJEUaQtppYqi3TExrxi2nl0nV4DFjhlSLTFiXMI6QfCEkRRpCypb/hjfJwXPTSwuPlasSBhKS\nL4SkSH0y/8yW/4bD+q+vX6yrSfriZkLyhZAUKfgwXacTGpcv6pwwkJB8ISRFCg5p2IANDcsj\nhycMJCRfCEmRgkO6SUbPTy+9NllmJAwkJF8ISZHCfwM7TaRHzbjxI3qLjEtKhZB8ISRFipjK\nMG9SrzYiraomPZM4jJB8ISRFipsTVPfxYmY2lAwhKVLk5DqmCJUQISlSREhMESoxQlKk8JCY\nIlRqhKRIEYe/mSJUYoSkSOG/kGWKUKkRkiJMEbKLkBRhipBdhKQIU4TsIiRFmCJkFyEpwhQh\nuwhJEaYI2UVIijBFyC5CUoQpQnYRkiJMEbKLkBRhipBdhKQIU4TsIiRF/EwR+vfOfRp8WdYW\nug0kIiRF/EwRWv/gvQ2u4BXJE0JShClCdhGSIkwRsouQFGGKkF2EpAhThOwiJEWYImQXISnC\nFCG7CEkRpgjZRUiKMEXILkJShClCdhGSIvwVIbsISRH+ipBdhKQIU4TsIiRFmCJkFyEpwhQh\nuwhJEaYI2UVIihQT0sevZo6Af/pBwihC8oWQFCk8pHm7i+xwe2rxkKS1EJIvhKRIwSG9WVE+\nakyF/CJaJqSSICRFCg5pYtkfg+CTfhWvBoRUIoSkSMEh9T4kOn2t3WEBIZUIISlScEhtT0md\nXSLPElKJEJIiBYfUa2DqbFWPPssJqTQISZGCQzpLpq+Izh+T8Z8RUkkQkiIFh7Skn5QfFC1c\nIpXdCKkUCEmRwn+PtPSCAel3dzP7CyGVAiEpUuQfP0mpe2d2wrWE5AshKeIipGSE5AshKUJI\ndhGSIoRkFyEpQkh2EZIihGQXISlCSHYRkiKEZBchKUJIdhGSIoRkFyEpQkh2EZIihGQXISlC\nSHYRkiKEZBchKUJIdhGSIoRkFyEpQkh2EZIihGQXISlCSHYRkiKEZBchKUJIdhGSIoRkFyEp\nQkh2EZIihGQXISlCSHYRkiKEZBchKUJIdhGSIoRkFyEpQkh2EZIihGQXISlCSHYRkiKEZBch\nKUJIdhGSIoRkFyEpQkh2EZIihGQXISlCSHYRkiKEZBchKUJIdhGSIoRkFyEpQkh2EZIihGQX\nISlCSHYRkiKEZBchKUJIdhGSIoRkFyEpQkh2EZIihGQXISlCSHYRkiKEZBchKUJIdhGSIoRk\nFyEpQkh2EZIihGQXISlCSHYRkiJFhrR0UW2+IYTkCyEpUkRIC6b0rRBpXT15TuIwQvKFkBQp\nPKTpZVI1eMyYIdUiE9YljCMkXwhJkYJDukkOnpteWnisXJUwkJB8ISRFCg5pWP/19Yt1NTUJ\nAwnJF0JSpOCQOp3QuHxR54SBhOQLISlS+CvSgA0NyyOHJwwkJF8ISZEiPiONnp9eem2yzEgY\nSEi+EJIihR+1mybSo2bc+BG9RcYlpUJIvhCSIkX8HmnepF5tRFpVTXomcRgh+UJIihQ3s6Hu\n48XMbCgZQlKEKUJ2EZIiTBGyi5AUYYqQXYSkCFOE7CIkRZgiZBchKcIUIbsISRGmCNlFSIow\nRcguQlKEKUJ2EZIiTBGyi5AUYYqQXYSkCFOE7CIkRZgiZBchKcIUIbsISRGmCNlFSIowRcgu\nQlLEzxShd3bo0qCjrC10G0hESIr4mSK04eF7G1zBK5InhKQIU4TsIiRFmCJkFyEpwhQhuwhJ\nEaYI2UVIijBFyC5CUoQpQnYRkiJMEbKLkBRhipBdhKQIU4TsIiRF+CtCdhGSIvwVIbsISRGm\nCNlFSIowRcguQlKEKUJ2EZIiTBGyi5AUKXKKUDMQki+EpAgh2UVIihCSXYSkCCHZRUiKFBrS\nz7fNkjCSkHwhJEUKDemNs9pKx90aJIwkJF8ISZHC39o9IWObNY6QfCEkRYr4jLQLIZUWISlS\nREiTJzRrGCH5QkiKcNTOLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDs\nIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ\n7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKE\nkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJS\nhJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5C\nUoSQ7CIkRQjJLkJShJDsIilAdP4AAAzhSURBVCRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJ\nLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUI\nyS5CUoSQ7CIkRQjJLkJShJDsIiRFCMkuQlKEkOwiJEUIyS5CUoSQ7CIkRQhJqRdm5fWN4/Ku\npTkhnVx9al5nfuLiPm3JCEmnV6QZdsm7muaEVNPm6LxaPebiTm3JCEmnl2Rp3jH9+uUd0qyQ\ntsu/O+0JKQ9C0omQjCEknQjJGELSiZCMKTKkpYtq8w0hpEIQkjFFhLRgSt8KkdbVk+ckDiOk\nQhCSMYWHNL1MqgaPGTOkWmTCuoRxhFQIbSHdsyyvpCfBlq/gkG6Sg+emlxYeK1clDCSkjdyZ\n/zegR8mneVfTgiGVNeO3WkPyr2YLVnBIw/qvr1+sq6lJGLglhfRF/n+X3/vJ1XntPDBvSF+X\nN/PuTQuGJKf+M59TOub/ve7Rc/I/gEvzD1mef4dbXMEhdTqhcfmizjlXvv2lLg06yqZe8y9v\nxj9zZZ275NO+bd4hXVp3yjukshn/6jbnH+bmaJ1/byT/HS8vzzuks1TmHdO6LO+QLtIu75A2\njh4bR5rzn+oHhT75m1D4K9KADQ3LI4fnXFn758YZYU/dualVLMo/nWzWL5/KO+TBO/Kv5ldP\n5B3y6K35V3Pro3mHPP6r/Ku548G8Q576Zf7V3HNP/jE8fgkWFfrkb0IRn5FGz08vvTZZZrja\nHcCmwo/aTRPpUTNu/IjeIuO2nE9BQEGK+D3SvEm9wvfFraomPeNudwCbipvZUPfx4rwzG4Ct\ngP+5dsBWgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJcICQAAcICXCAkAAHCAlwgJAABwgJ\ncICQAAcICXBgCwjpGO9/2glbJpd/0nILCOn8/fL+8cIWdH27Uu9B3NPy+1LvQtxOF5V6D+IO\nm+TwWbgFhKTry4Qf7VDqPYhbKi+Vehfi+t9c6j2Ia8aXWTcfITlGSAkISTNC2jRCSkBI2Qhp\n0wgpASFlI6RNI6QEhJSNkDaNkBIQUjZC2jRCSkBI2Qhp0wgpASFlI6RNI6QEhJSNkDaNkBIQ\nUrbLx5R6D+Ke6lLqPYhbUfZKqXchbtBtpd6DuFNPdbiyLSCkVR+Veg/iat8p9R5keavUO5Dl\nfVXfSLdsmcOVbQEhAaVHSIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOE\nBDhASIADhAQ4YC+klTPfL/UuwBr/Txp7IU2VR5Ou/sXwzsN/kV588qBOVce86W9Plp7Tp6LP\ncYkbaLG9iW0o/4j8g4uj6YFJ8/+kMRfSfZL4mEyT/lN2kenR4s+katK4Vl3f9bUnS/vJwFO+\nUdZunoK9iW0o/4j8g4uj6YFJa4EnjbWQPtiuMukxmSeHrA/Wf6NsQRC823rw8iB4RE7wtSsX\nyhnh6WPle5R+b2Ibyj8i/+AiKXpg0lriSWMspLoDe1+Ufkw2XDW0steZi3Oun5T6qzn/ih6I\nc+WF6JKf/9LXvgwoXxGdjZKPS743sQ3l35X4YC8UPTApLfKkMRbSNeV/vTr1mHyxv+xz6gHS\nM/MafM3j6fNe1amzqr7hYk/P+zLwoNTZGHm15HsT21AzdqVx0Q9FD0x6qy3xpLEV0rw2Fwbp\nx+R6uSI8vV2OTF/R74LUWV2b4anzwRVB0PZrLx62fY+j3vC8S59UdF9f6r2JbyjvrsQH+6Th\ngUlpmSeNqZDWDNzzi8xj0rNfbXTJ0DarU9dkHpOPZVzqfIwsXSZ9KweddEh5+3943aXX+slv\nSr43sQ3l35X4YI9UPDCRFnrSmAhp9fWh8JE4o2JhkH5MVsnQOyMjZX5qROYxWSzjU+djZNF7\nIhfWBcGssq/62pvQqkvbVdxY2r3J2VD+XYkt+tiXNCUPTKSFnjQmQvoo+grqo4LZcl2QeUxe\nbvhm6ueDW0JfGh2ezA5qW41I3WBI69q10m1DtPiN6COvj70J/bGnjH01KO3eRGIbyr8rsUUP\nu5Km5YEJtdSTxkRIGdc2PBC3LEkdYU1ruDR8clf1SV3UI/z42GWf1OI0+Zev/blU+jybWij5\n3jRuqBm7Elv0RM8D03JPGkshzZoWGSyjp80Juqbv8Q2Xpq/KvEoHx8lr4elCmRQEh3T6PLpk\nRPkqT7szU8YvzyyWem9iG8q/K7FFPxQ9MC33pLEUUlr6c+P35crw9MmyzLOh/jF5Rr4ZBHXH\nyl/D62R6+N7lHhnraT/q+ld+Vr9c6r2JbSj/rsQWvdD0wNTz/6SxGtKK3WTv08e32vGD9IX1\nj0kwVg68aIQcnl4cdOrXpcrXbMV3pNshaZ+Wfm8aN9SsXWlY9EHVA5Ph/0ljNaRgzfl7te87\nrf7AU8NjsvbyoZ2GzUh96cHaGTUdB57h8hsHsvyp4W32B6Xfm9iGmrErjYs+qHpgMvw/aeyF\nBChESIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiA\nA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhA\nSIADhLQlusXz93ZhI4RkVU11EBze5H++upfkd9F3Cs+Rr2Uu+Ysckjp/4+cttHNbH0KyauOQ\nnur1UHR29w4iUnFVXVDXs+zD9DXflttS52dvm/757eP6td/tvOUBnCEkq6KQlnwQv+QPcmd4\n+piMekYeOFl+EQTnyc/S1/TaJvVe76m26ZDe6ND6wGmD5Suft+web9EISaPmfFFyFFK2dEgj\nqj8PZGHdnrsEwYsyPHXFXDk0PJ3cXyQd0pHySBC9TPFGzx1CKqElJ/bsOXVJt1PCxQ1XDa3s\ndebicOmUbd87oKzNbrcGWZfGvX5U9Y5HvxOFdFT0n++3g7ftuv8TQTAq+iLxJcEOo8P/qguD\n20+qC4Jdy1Lfc3+J3B6eThg7tmM6pO59otP5MrWl7uhWgJBKZ3Hf1odO6d67fRjSF/vLPqce\nID3fDUPqMKjXOdM7yf1Zl8Y836nsgON77LBTJqQrpeq4KZ3Lnw2ePEtOnbk2GFy1Kgop5Yfy\n0+jsK20yH4d2S4VUe+PD0dksOaOF7ujWgJBK50x5KAgW9ZIwpOvlivCC2+XIMCQZ9Fl0xG1i\n1qUx+5Y/GAQraiQTUrf+a4PgBTmp/q3dT2XfJ+tDekOGhKevy7jMTXfbtn4lKxc8uHPlgpa4\nl1sJQiqZtRXDorOfRSH17FcbLQ9tszoM6d5osXJU1qWN/iHHRGd/z4S0rlXvdeGPC9+pDyn4\nToXIoJ9+lBq8r4SvZlenLw/iIU0TaTfL693byhBSybwi343O/h6GtEqG3hkZKfPDkN6MLu42\nKuvSRndmDmV3y7wiHS49L/pTqrRMSMGy66S9tPl1tPhTuTYIBlesyNy2MaQX75nRo+0ffN/F\nrQghlcyT8uPo7N9hSC9LvefDkJZEF4chxS9tdK08njr/aiakNZf3EWk/+cPGkML/qnPv2r5V\n9Mnqw/L9gg/KJtTftjGk6LqOO3q9f1sXQiqZF+V70dk/w5CWxD73N4a0pMmjAffIb1LnveqP\n2gXBa7fuL4Pq0iGtfXV56qjdvemqRso7N8rd9bdNh/TmzekPRyOFmUTOEFLJrCofEZ3dFH1G\n6rpP6qIbLo2HFL+00dzoMEQQvF2eDumNi5+Kfvx6+GEoFdJb8pNUSHPlmujyX8uPD2y3qv62\n6ZCek7PSP1XWer2DWxVCKp2T5LEg+KhPFNL35crwgifLJmWFFLs0Zkj5I0Hw+aGZgw1vyR7r\ngmDdPm2/CEMKPz19vs34VEjXSiqwZW36tz6q4abpkNZt3/mt8OxuGd9Cd3RrQEil81F1mwkn\nV02S7wTBit1k79PHt9rxg6yQYpfGPN+p/Osn962s/4XsOOl36mk7yyXRh659rlgdnCUXLJF5\n93UYtCE1+jCR+xpumvmMdE9Z+6NPHynds9eLYhBSCX08qfuuP1woV4eLa87fq33faYuCrJBi\nl8a9dmSPHY6YOy0T0oof7Nqh2/Df1YUvRmMrtlsWrDxBWsk28tW30oN/Lx0aD57XH2x4+pCu\n7fc4l09IDhFS6fzj5ej0sczBA3feuEMufc7xOpEHIZXO0K4rg6Du8HafOl9z/cwGtBhCKp1H\nyva85scHyXnu13zBR+7XiUSEVEJP7r9dtwPuaMbAa6XRd73vFgpASIADhAQ4QEiAA4QEOEBI\ngAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA4QEOEBIgAOEBDhASIADhAQ4\nQEiAA4QEOEBIgAOEBDhASIADhAQ4QEiAA/8fLjnny7ylDYUAAAAASUVORK5CYII=",
      "text/plain": [
       "Plot with title “Histogram of gene_dist$V13”"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "hist(gene_dist$V13)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "9177"
      ],
      "text/latex": [
       "9177"
      ],
      "text/markdown": [
       "9177"
      ],
      "text/plain": [
       "[1] 9177"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td> 11873</td><td> 14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>chr1</td><td> 17368</td><td> 17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>2960</td></tr>\n",
       "\t<tr><td>chr1</td><td> 17368</td><td> 17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>chr1</td><td> 11873</td><td> 14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>2960</td></tr>\n",
       "\t<tr><td>chr1</td><td> 30365</td><td> 30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>chr1</td><td> 34610</td><td> 36081</td><td>FAM138A:NR_026818.1  </td><td>.</td><td>-</td><td>4108</td></tr>\n",
       "\t<tr><td>chr1</td><td> 34610</td><td> 36081</td><td>FAM138A:NR_026818.1  </td><td>.</td><td>-</td><td>chr1</td><td> 30365</td><td> 30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>4108</td></tr>\n",
       "\t<tr><td>chr1</td><td>182387</td><td>184878</td><td>DDX11L17:NR_148357.1 </td><td>.</td><td>+</td><td>chr1</td><td>187890</td><td>187958</td><td>MIR6859-2:NR_107062.1</td><td>.</td><td>-</td><td>3013</td></tr>\n",
       "\t<tr><td>chr1</td><td>187890</td><td>187958</td><td>MIR6859-2:NR_107062.1</td><td>.</td><td>-</td><td>chr1</td><td>182387</td><td>184878</td><td>DDX11L17:NR_148357.1 </td><td>.</td><td>+</td><td>3013</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 &  11873 &  14409 & DDX11L1:NR\\_046018.2   & . & + & chr1 &  17368 &  17436 & MIR6859-1:NR\\_106918.1 & . & - & 2960\\\\\n",
       "\t chr1 &  17368 &  17436 & MIR6859-1:NR\\_106918.1 & . & - & chr1 &  11873 &  14409 & DDX11L1:NR\\_046018.2   & . & + & 2960\\\\\n",
       "\t chr1 &  30365 &  30503 & MIR1302-2:NR\\_036051.1 & . & + & chr1 &  34610 &  36081 & FAM138A:NR\\_026818.1   & . & - & 4108\\\\\n",
       "\t chr1 &  34610 &  36081 & FAM138A:NR\\_026818.1   & . & - & chr1 &  30365 &  30503 & MIR1302-2:NR\\_036051.1 & . & + & 4108\\\\\n",
       "\t chr1 & 182387 & 184878 & DDX11L17:NR\\_148357.1  & . & + & chr1 & 187890 & 187958 & MIR6859-2:NR\\_107062.1 & . & - & 3013\\\\\n",
       "\t chr1 & 187890 & 187958 & MIR6859-2:NR\\_107062.1 & . & - & chr1 & 182387 & 184878 & DDX11L17:NR\\_148357.1  & . & + & 3013\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 13\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | V11 &lt;chr&gt; | V12 &lt;chr&gt; | V13 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 |  11873 |  14409 | DDX11L1:NR_046018.2   | . | + | chr1 |  17368 |  17436 | MIR6859-1:NR_106918.1 | . | - | 2960 |\n",
       "| chr1 |  17368 |  17436 | MIR6859-1:NR_106918.1 | . | - | chr1 |  11873 |  14409 | DDX11L1:NR_046018.2   | . | + | 2960 |\n",
       "| chr1 |  30365 |  30503 | MIR1302-2:NR_036051.1 | . | + | chr1 |  34610 |  36081 | FAM138A:NR_026818.1   | . | - | 4108 |\n",
       "| chr1 |  34610 |  36081 | FAM138A:NR_026818.1   | . | - | chr1 |  30365 |  30503 | MIR1302-2:NR_036051.1 | . | + | 4108 |\n",
       "| chr1 | 182387 | 184878 | DDX11L17:NR_148357.1  | . | + | chr1 | 187890 | 187958 | MIR6859-2:NR_107062.1 | . | - | 3013 |\n",
       "| chr1 | 187890 | 187958 | MIR6859-2:NR_107062.1 | . | - | chr1 | 182387 | 184878 | DDX11L17:NR_148357.1  | . | + | 3013 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2     V3     V4                    V5 V6 V7   V8     V9    \n",
       "1 chr1  11873  14409 DDX11L1:NR_046018.2   .  +  chr1  17368  17436\n",
       "2 chr1  17368  17436 MIR6859-1:NR_106918.1 .  -  chr1  11873  14409\n",
       "3 chr1  30365  30503 MIR1302-2:NR_036051.1 .  +  chr1  34610  36081\n",
       "4 chr1  34610  36081 FAM138A:NR_026818.1   .  -  chr1  30365  30503\n",
       "5 chr1 182387 184878 DDX11L17:NR_148357.1  .  +  chr1 187890 187958\n",
       "6 chr1 187890 187958 MIR6859-2:NR_107062.1 .  -  chr1 182387 184878\n",
       "  V10                   V11 V12 V13 \n",
       "1 MIR6859-1:NR_106918.1 .   -   2960\n",
       "2 DDX11L1:NR_046018.2   .   +   2960\n",
       "3 FAM138A:NR_026818.1   .   -   4108\n",
       "4 MIR1302-2:NR_036051.1 .   +   4108\n",
       "5 MIR6859-2:NR_107062.1 .   -   3013\n",
       "6 DDX11L17:NR_148357.1  .   +   3013"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 × 13</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th><th scope=col>V12</th><th scope=col>V13</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chrY</td><td> 1386151</td><td> 1392113</td><td>SLC25A6_1:NM_001636.4_1           </td><td>.</td><td>-</td><td>chrY</td><td> 1336784</td><td> 1382689</td><td>IL3RA_1:NM_002183.4_1,NM_001267713.2_1,XM_005274780.6,XM_005274781.2,XM_017030043.3</td><td>.</td><td>+</td><td>3463</td></tr>\n",
       "\t<tr><td>chrY</td><td> 1397024</td><td> 1399412</td><td>LINC00106_1:NR_130733.1_1         </td><td>.</td><td>+</td><td>chrY</td><td> 1403138</td><td> 1452909</td><td>ASMTL_1:NM_004192.4_1,NM_001173474.2_1,XM_047442774.1                              </td><td>.</td><td>-</td><td>3727</td></tr>\n",
       "\t<tr><td>chrY</td><td> 1397024</td><td> 1399412</td><td>LINC00106_1:NR_130733.1_1         </td><td>.</td><td>+</td><td>chrY</td><td> 1403138</td><td> 1453756</td><td>ASMTL_1:NM_001173473.2_1,XM_005274783.4,XM_047442773.1                             </td><td>.</td><td>-</td><td>3727</td></tr>\n",
       "\t<tr><td>chrY</td><td> 2609190</td><td> 2609254</td><td>MIR6089_1:NR_106737.1_1           </td><td>.</td><td>+</td><td>chrY</td><td> 2612990</td><td> 2615347</td><td>LINC00102_1:NR_037842.1_1                                                          </td><td>.</td><td>-</td><td>3737</td></tr>\n",
       "\t<tr><td>chrY</td><td>57184215</td><td>57197337</td><td>IL9R_1:NM_002186.3_1,NM_176786.2_1</td><td>.</td><td>+</td><td>chrY</td><td>57201083</td><td>57203350</td><td>WASIR1_1:NR_138048.1_1                                                             </td><td>.</td><td>-</td><td>3747</td></tr>\n",
       "\t<tr><td>chrY</td><td>57201083</td><td>57203350</td><td>WASIR1_1:NR_138048.1_1            </td><td>.</td><td>-</td><td>chrY</td><td>57184215</td><td>57197337</td><td>IL9R_1:NM_002186.3_1,NM_176786.2_1                                                 </td><td>.</td><td>+</td><td>3747</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 × 13\n",
       "\\begin{tabular}{lllllllllllll}\n",
       " V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11 & V12 & V13\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t chrY &  1386151 &  1392113 & SLC25A6\\_1:NM\\_001636.4\\_1            & . & - & chrY &  1336784 &  1382689 & IL3RA\\_1:NM\\_002183.4\\_1,NM\\_001267713.2\\_1,XM\\_005274780.6,XM\\_005274781.2,XM\\_017030043.3 & . & + & 3463\\\\\n",
       "\t chrY &  1397024 &  1399412 & LINC00106\\_1:NR\\_130733.1\\_1          & . & + & chrY &  1403138 &  1452909 & ASMTL\\_1:NM\\_004192.4\\_1,NM\\_001173474.2\\_1,XM\\_047442774.1                               & . & - & 3727\\\\\n",
       "\t chrY &  1397024 &  1399412 & LINC00106\\_1:NR\\_130733.1\\_1          & . & + & chrY &  1403138 &  1453756 & ASMTL\\_1:NM\\_001173473.2\\_1,XM\\_005274783.4,XM\\_047442773.1                              & . & - & 3727\\\\\n",
       "\t chrY &  2609190 &  2609254 & MIR6089\\_1:NR\\_106737.1\\_1            & . & + & chrY &  2612990 &  2615347 & LINC00102\\_1:NR\\_037842.1\\_1                                                           & . & - & 3737\\\\\n",
       "\t chrY & 57184215 & 57197337 & IL9R\\_1:NM\\_002186.3\\_1,NM\\_176786.2\\_1 & . & + & chrY & 57201083 & 57203350 & WASIR1\\_1:NR\\_138048.1\\_1                                                              & . & - & 3747\\\\\n",
       "\t chrY & 57201083 & 57203350 & WASIR1\\_1:NR\\_138048.1\\_1             & . & - & chrY & 57184215 & 57197337 & IL9R\\_1:NM\\_002186.3\\_1,NM\\_176786.2\\_1                                                  & . & + & 3747\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 × 13\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; | V5 &lt;chr&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;int&gt; | V9 &lt;int&gt; | V10 &lt;chr&gt; | V11 &lt;chr&gt; | V12 &lt;chr&gt; | V13 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chrY |  1386151 |  1392113 | SLC25A6_1:NM_001636.4_1            | . | - | chrY |  1336784 |  1382689 | IL3RA_1:NM_002183.4_1,NM_001267713.2_1,XM_005274780.6,XM_005274781.2,XM_017030043.3 | . | + | 3463 |\n",
       "| chrY |  1397024 |  1399412 | LINC00106_1:NR_130733.1_1          | . | + | chrY |  1403138 |  1452909 | ASMTL_1:NM_004192.4_1,NM_001173474.2_1,XM_047442774.1                               | . | - | 3727 |\n",
       "| chrY |  1397024 |  1399412 | LINC00106_1:NR_130733.1_1          | . | + | chrY |  1403138 |  1453756 | ASMTL_1:NM_001173473.2_1,XM_005274783.4,XM_047442773.1                              | . | - | 3727 |\n",
       "| chrY |  2609190 |  2609254 | MIR6089_1:NR_106737.1_1            | . | + | chrY |  2612990 |  2615347 | LINC00102_1:NR_037842.1_1                                                           | . | - | 3737 |\n",
       "| chrY | 57184215 | 57197337 | IL9R_1:NM_002186.3_1,NM_176786.2_1 | . | + | chrY | 57201083 | 57203350 | WASIR1_1:NR_138048.1_1                                                              | . | - | 3747 |\n",
       "| chrY | 57201083 | 57203350 | WASIR1_1:NR_138048.1_1             | . | - | chrY | 57184215 | 57197337 | IL9R_1:NM_002186.3_1,NM_176786.2_1                                                  | . | + | 3747 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2       V3       V4                                 V5 V6 V7   V8      \n",
       "1 chrY  1386151  1392113 SLC25A6_1:NM_001636.4_1            .  -  chrY  1336784\n",
       "2 chrY  1397024  1399412 LINC00106_1:NR_130733.1_1          .  +  chrY  1403138\n",
       "3 chrY  1397024  1399412 LINC00106_1:NR_130733.1_1          .  +  chrY  1403138\n",
       "4 chrY  2609190  2609254 MIR6089_1:NR_106737.1_1            .  +  chrY  2612990\n",
       "5 chrY 57184215 57197337 IL9R_1:NM_002186.3_1,NM_176786.2_1 .  +  chrY 57201083\n",
       "6 chrY 57201083 57203350 WASIR1_1:NR_138048.1_1             .  -  chrY 57184215\n",
       "  V9      \n",
       "1  1382689\n",
       "2  1452909\n",
       "3  1453756\n",
       "4  2615347\n",
       "5 57203350\n",
       "6 57197337\n",
       "  V10                                                                                \n",
       "1 IL3RA_1:NM_002183.4_1,NM_001267713.2_1,XM_005274780.6,XM_005274781.2,XM_017030043.3\n",
       "2 ASMTL_1:NM_004192.4_1,NM_001173474.2_1,XM_047442774.1                              \n",
       "3 ASMTL_1:NM_001173473.2_1,XM_005274783.4,XM_047442773.1                             \n",
       "4 LINC00102_1:NR_037842.1_1                                                          \n",
       "5 WASIR1_1:NR_138048.1_1                                                             \n",
       "6 IL9R_1:NM_002186.3_1,NM_176786.2_1                                                 \n",
       "  V11 V12 V13 \n",
       "1 .   +   3463\n",
       "2 .   -   3727\n",
       "3 .   -   3727\n",
       "4 .   -   3737\n",
       "5 .   -   3747\n",
       "6 .   +   3747"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_dist_downstream_15kb_dt <- subset(gene_dist_other_overlaps, V13 >0 & V13 < 7500)\n",
    "nrow(gene_dist_downstream_15kb_dt)\n",
    "head(gene_dist_downstream_15kb_dt)\n",
    "tail(gene_dist_downstream_15kb_dt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "7350"
      ],
      "text/latex": [
       "7350"
      ],
      "text/markdown": [
       "7350"
      ],
      "text/plain": [
       "[1] 7350"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_dist_downstream_15kb <- unique(c(gene_dist_downstream_15kb_dt$V4, \n",
    "                                     gene_dist_downstream_15kb_dt$V10))\n",
    "length(gene_dist_downstream_15kb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "16569"
      ],
      "text/latex": [
       "16569"
      ],
      "text/markdown": [
       "16569"
      ],
      "text/plain": [
       "[1] 16569"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "overlap_genes <- unique(c(genes_overlap_opposite_strands, \n",
    "                         gene_dist_downstream_15kb))\n",
    "length(overlap_genes)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter bidirectionals"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get bidirectionals on opposite strands of overapping genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gene_chrom</th><th scope=col>gene_start</th><th scope=col>gene_end</th><th scope=col>gene_id</th><th scope=col>gene_score</th><th scope=col>gene_strand</th><th scope=col>overlap_len</th><th scope=col>bidir_width</th><th scope=col>bidir_overlap_perc</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>2368</td><td>17686</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td> 410</td><td> 410</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & gene\\_chrom & gene\\_start & gene\\_end & gene\\_id & gene\\_score & gene\\_strand & overlap\\_len & bidir\\_width & bidir\\_overlap\\_perc\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 & 2368 & 17686 & MIR6859-1:NR\\_106918.1 & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - &  410 &  410 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 15\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gene_chrom &lt;chr&gt; | gene_start &lt;int&gt; | gene_end &lt;int&gt; | gene_id &lt;chr&gt; | gene_score &lt;chr&gt; | gene_strand &lt;chr&gt; | overlap_len &lt;int&gt; | bidir_width &lt;dbl&gt; | bidir_overlap_perc &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 | 2368 | 17686 | MIR6859-1:NR_106918.1 | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - |  410 |  410 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand gene_chrom gene_start gene_end\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      chr1          0       29620   \n",
       "2 chr1  3917  4919 chr1:3917-4919 14    .      chr1       2368       17686   \n",
       "3 chr1  5632  6042 chr1:5632-6042 14    .      chr1          0       29620   \n",
       "  gene_id               gene_score gene_strand overlap_len bidir_width\n",
       "1 WASH7P:NR_024540.1    .          -           1002        1002       \n",
       "2 MIR6859-1:NR_106918.1 .          -           1002        1002       \n",
       "3 WASH7P:NR_024540.1    .          -            410         410       \n",
       "  bidir_overlap_perc\n",
       "1 1                 \n",
       "2 1                 \n",
       "3 1                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(bidir_overlaps, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "529446"
      ],
      "text/latex": [
       "529446"
      ],
      "text/markdown": [
       "529446"
      ],
      "text/plain": [
       "[1] 529446"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1:1000004-1000574    </td><td>1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1:100001920-100002206</td><td>1</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1:100002946-100003218</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & chr1:1000004-1000574     & 1\\\\\n",
       "\t2 & chr1:100001920-100002206 & 1\\\\\n",
       "\t3 & chr1:100002946-100003218 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 1 | chr1:1000004-1000574     | 1 |\n",
       "| 2 | chr1:100001920-100002206 | 1 |\n",
       "| 3 | chr1:100002946-100003218 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Var1                     Freq\n",
       "1 chr1:1000004-1000574     1   \n",
       "2 chr1:100001920-100002206 1   \n",
       "3 chr1:100002946-100003218 1   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#filter regions where bidirectional overlaps genes + 15kb downstream PAS plus 250bp upstream TSS\n",
    "overlaps_high <- subset(bidir_overlaps, bidir_overlap_perc > 0 ) ##keep this if you want to filter based on %overlap\n",
    "\n",
    "#get unique bidirectional transcripts by strand\n",
    "#meaning, if a bidirectional is overlapping more than 50% of two genes + 30kb dPAS\n",
    "#it is kept for both instances\n",
    "overlaps_bidirs_unique <- unique(overlaps_high[,c('bidirs','gene_strand')])\n",
    "\n",
    "#count to get bidirectional transcripts appearing based on strand with table()\n",
    "#`unique` will return all bidirectionals near same strand transcripts (Freq==1)\n",
    "#OR bidirectional transcripts near two opposite strand genes (Freq==2)\n",
    "overlaps_bidirs_counts <- as.data.frame(table(overlaps_bidirs_unique$bidirs))\n",
    "\n",
    "#get bidirectionals not overlapping two opposite strands\n",
    "overlaps_bidirs_non_multiple <- subset(overlaps_bidirs_counts, Freq < 2)\n",
    "nrow(overlaps_bidirs_non_multiple)\n",
    "head(overlaps_bidirs_non_multiple, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "59730"
      ],
      "text/latex": [
       "59730"
      ],
      "text/markdown": [
       "59730"
      ],
      "text/plain": [
       "[1] 59730"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>53</th><td>chr1:100069757-100070077</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>54</th><td>chr1:100074130-100074330</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>55</th><td>chr1:100074535-100074769</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t53 & chr1:100069757-100070077 & 2\\\\\n",
       "\t54 & chr1:100074130-100074330 & 2\\\\\n",
       "\t55 & chr1:100074535-100074769 & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 53 | chr1:100069757-100070077 | 2 |\n",
       "| 54 | chr1:100074130-100074330 | 2 |\n",
       "| 55 | chr1:100074535-100074769 | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "   Var1                     Freq\n",
       "53 chr1:100069757-100070077 2   \n",
       "54 chr1:100074130-100074330 2   \n",
       "55 chr1:100074535-100074769 2   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>589169</th><td>chrY:9494767-9495735</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>589174</th><td>chrY:9703970-9704450</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>589175</th><td>chrY:9765870-9766390</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & Var1 & Freq\\\\\n",
       "  & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t589169 & chrY:9494767-9495735 & 2\\\\\n",
       "\t589174 & chrY:9703970-9704450 & 2\\\\\n",
       "\t589175 & chrY:9765870-9766390 & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 2\n",
       "\n",
       "| <!--/--> | Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| 589169 | chrY:9494767-9495735 | 2 |\n",
       "| 589174 | chrY:9703970-9704450 | 2 |\n",
       "| 589175 | chrY:9765870-9766390 | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "       Var1                 Freq\n",
       "589169 chrY:9494767-9495735 2   \n",
       "589174 chrY:9703970-9704450 2   \n",
       "589175 chrY:9765870-9766390 2   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#number of transcripts in overlapping genes\n",
    "overlaps_bidirs_multiple <- subset(overlaps_bidirs_counts, Freq >= 2)\n",
    "nrow(overlaps_bidirs_multiple)\n",
    "head(overlaps_bidirs_multiple, 3)\n",
    "tail(overlaps_bidirs_multiple, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "589176"
      ],
      "text/latex": [
       "589176"
      ],
      "text/markdown": [
       "589176"
      ],
      "text/plain": [
       "[1] 589176"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##check the number of overlaps\n",
    "nrow(overlaps_bidirs_non_multiple) + nrow(overlaps_bidirs_multiple)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "979466"
      ],
      "text/latex": [
       "979466"
      ],
      "text/markdown": [
       "979466"
      ],
      "text/plain": [
       "[1] 979466"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>gene_chrom</th><th scope=col>gene_start</th><th scope=col>gene_end</th><th scope=col>gene_id</th><th scope=col>gene_score</th><th scope=col>gene_strand</th><th scope=col>overlap_len</th><th scope=col>bidir_width</th><th scope=col>bidir_overlap_perc</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>chr1</td><td>2368</td><td>17686</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>1002</td><td>1002</td><td>1</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>chr1</td><td>   0</td><td>29620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td> 410</td><td> 410</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 15\n",
       "\\begin{tabular}{lllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & gene\\_chrom & gene\\_start & gene\\_end & gene\\_id & gene\\_score & gene\\_strand & overlap\\_len & bidir\\_width & bidir\\_overlap\\_perc\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & chr1 & 2368 & 17686 & MIR6859-1:NR\\_106918.1 & . & - & 1002 & 1002 & 1\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & chr1 &    0 & 29620 & WASH7P:NR\\_024540.1    & . & - &  410 &  410 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 15\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | gene_chrom &lt;chr&gt; | gene_start &lt;int&gt; | gene_end &lt;int&gt; | gene_id &lt;chr&gt; | gene_score &lt;chr&gt; | gene_strand &lt;chr&gt; | overlap_len &lt;int&gt; | bidir_width &lt;dbl&gt; | bidir_overlap_perc &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | chr1 | 2368 | 17686 | MIR6859-1:NR_106918.1 | . | - | 1002 | 1002 | 1 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | chr1 |    0 | 29620 | WASH7P:NR_024540.1    | . | - |  410 |  410 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand gene_chrom gene_start gene_end\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      chr1          0       29620   \n",
       "2 chr1  3917  4919 chr1:3917-4919 14    .      chr1       2368       17686   \n",
       "3 chr1  5632  6042 chr1:5632-6042 14    .      chr1          0       29620   \n",
       "  gene_id               gene_score gene_strand overlap_len bidir_width\n",
       "1 WASH7P:NR_024540.1    .          -           1002        1002       \n",
       "2 MIR6859-1:NR_106918.1 .          -           1002        1002       \n",
       "3 WASH7P:NR_024540.1    .          -            410         410       \n",
       "  bidir_overlap_perc\n",
       "1 1                 \n",
       "2 1                 \n",
       "3 1                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#get bidirectionals not overlapping multiple genes and dPAS\n",
    "overlaps_non_multiple <- overlaps_high[overlaps_high$bidirs %in% \n",
    "                                  as.character(overlaps_bidirs_non_multiple$Var1),]\n",
    "nrow(overlaps_non_multiple)\n",
    "head(overlaps_non_multiple, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "529446"
      ],
      "text/latex": [
       "529446"
      ],
      "text/markdown": [
       "529446"
      ],
      "text/plain": [
       "[1] 529446"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(overlaps_non_multiple$bidirs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "260012"
      ],
      "text/latex": [
       "260012"
      ],
      "text/markdown": [
       "260012"
      ],
      "text/plain": [
       "[1] 260012"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>3917</td><td>4919</td><td>chr1:3917-4919</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>5632</td><td>6042</td><td>chr1:5632-6042</td><td>14</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>6132</td><td>6486</td><td>chr1:6132-6486</td><td> 7</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 3917 & 4919 & chr1:3917-4919 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 5632 & 6042 & chr1:5632-6042 & 14 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 6132 & 6486 & chr1:6132-6486 &  7 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 3917 | 4919 | chr1:3917-4919 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 5632 | 6042 | chr1:5632-6042 | 14 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 6132 | 6486 | chr1:6132-6486 |  7 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end  bidirs         score strand SRR7266931 SRR7266932 SRR7266933\n",
       "1 chr1  3917  4919 chr1:3917-4919 14    .      0          0          0         \n",
       "2 chr1  5632  6042 chr1:5632-6042 14    .      0          0          0         \n",
       "3 chr1  6132  6486 chr1:6132-6486  7    .      0          0          0         \n",
       "  SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719 SRR8483105\n",
       "1 0          ⋯ 0          0          0          0          0         \n",
       "2 0          ⋯ 0          0          0          0          0         \n",
       "3 0          ⋯ 0          0          0          0          0         \n",
       "  SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0         \n",
       "2 0          0          0          0          0         \n",
       "3 0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#get sense counts for bidirectionals overlapping the negative strand\n",
    "counts_pos_bidirs_overlaps <- get_strand_counts(counts_pos_bidirs,\n",
    "                                                overlaps_non_multiple,\n",
    "                                                \"-\")\n",
    "nrow(counts_pos_bidirs_overlaps)\n",
    "head(counts_pos_bidirs_overlaps, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "269434"
      ],
      "text/latex": [
       "269434"
      ],
      "text/markdown": [
       "269434"
      ],
      "text/plain": [
       "[1] 269434"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>36343</td><td>36775</td><td>chr1:36343-36775</td><td>58</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>38375</td><td>38929</td><td>chr1:38375-38929</td><td>58</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>41385</td><td>41765</td><td>chr1:41385-41765</td><td> 2</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 36343 & 36775 & chr1:36343-36775 & 58 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 38375 & 38929 & chr1:38375-38929 & 58 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 41385 & 41765 & chr1:41385-41765 &  2 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 36343 | 36775 | chr1:36343-36775 | 58 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 38375 | 38929 | chr1:38375-38929 | 58 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 41385 | 41765 | chr1:41385-41765 |  2 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end   bidirs           score strand SRR7266931 SRR7266932\n",
       "1 chr1  36343 36775 chr1:36343-36775 58    .      0          0         \n",
       "2 chr1  38375 38929 chr1:38375-38929 58    .      0          0         \n",
       "3 chr1  41385 41765 chr1:41385-41765  2    .      0          0         \n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1 0          0          ⋯ 0          0          0          0         \n",
       "2 0          0          ⋯ 0          0          0          0         \n",
       "3 0          0          ⋯ 0          0          0          0         \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0          0         \n",
       "2 0          0          0          0          0          0         \n",
       "3 0          0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#get antisense counts for bidirectionals overlapping the positive strand\n",
    "counts_neg_bidirs_overlaps <- get_strand_counts(counts_neg_bidirs,\n",
    "                                                overlaps_non_multiple,\n",
    "                                                \"+\")\n",
    "nrow(counts_neg_bidirs_overlaps)\n",
    "head(counts_neg_bidirs_overlaps, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get non-gene-overlapping bidirectionals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "318075"
      ],
      "text/latex": [
       "318075"
      ],
      "text/markdown": [
       "318075"
      ],
      "text/plain": [
       "[1] 318075"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>12182</td><td>12456</td><td>chr1:12182-12456</td><td> 1</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>13264</td><td>13506</td><td>chr1:13264-13506</td><td>14</td><td>.</td><td>4</td><td>3</td><td>8</td><td>6</td><td>⋯</td><td>11</td><td>9</td><td>5</td><td>4</td><td>2</td><td>2</td><td>4</td><td>2</td><td>0</td><td>6</td></tr>\n",
       "\t<tr><td>chr1</td><td>14655</td><td>14811</td><td>chr1:14655-14811</td><td>14</td><td>.</td><td>1</td><td>0</td><td>0</td><td>1</td><td>⋯</td><td> 1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 12182 & 12456 & chr1:12182-12456 &  1 & . & 0 & 0 & 0 & 0 & ⋯ &  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 13264 & 13506 & chr1:13264-13506 & 14 & . & 4 & 3 & 8 & 6 & ⋯ & 11 & 9 & 5 & 4 & 2 & 2 & 4 & 2 & 0 & 6\\\\\n",
       "\t chr1 & 14655 & 14811 & chr1:14655-14811 & 14 & . & 1 & 0 & 0 & 1 & ⋯ &  1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 12182 | 12456 | chr1:12182-12456 |  1 | . | 0 | 0 | 0 | 0 | ⋯ |  0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 13264 | 13506 | chr1:13264-13506 | 14 | . | 4 | 3 | 8 | 6 | ⋯ | 11 | 9 | 5 | 4 | 2 | 2 | 4 | 2 | 0 | 6 |\n",
       "| chr1 | 14655 | 14811 | chr1:14655-14811 | 14 | . | 1 | 0 | 0 | 1 | ⋯ |  1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end   bidirs           score strand SRR7266931 SRR7266932\n",
       "1 chr1  12182 12456 chr1:12182-12456  1    .      0          0         \n",
       "2 chr1  13264 13506 chr1:13264-13506 14    .      4          3         \n",
       "3 chr1  14655 14811 chr1:14655-14811 14    .      1          0         \n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1 0          0          ⋯  0         0          0          0         \n",
       "2 8          6          ⋯ 11         9          5          4         \n",
       "3 0          1          ⋯  1         0          0          0         \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0          0         \n",
       "2 2          2          4          2          0          6         \n",
       "3 0          0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#######################################################################################################\n",
    "##remove bidirectional transcripts overlapping genes\n",
    "##i: overlapping a single gene transcript region : overlaps_non_multiple, and\n",
    "##ii: overlapping +/- gene transcripts (these are removed) : overlaps_bidirs_multiple\n",
    "#counts_bidirs_nooverlaps <- counts_bidirs_pos_neg[!counts_bidirs_pos_neg$bidirs\n",
    "#                                                 %in%\n",
    "#                                                  unique(c(overlaps_non_multiple$bidirs, \n",
    "#                                                    as.character(overlaps_bidirs_multiple$Var1))),]\n",
    "#nrow(counts_bidirs_nooverlaps)\n",
    "#head(counts_bidirs_nooverlaps, 3)\n",
    "#######################################################################################################\n",
    "\n",
    "##Keep the transcripts overlapping multiple gene and filter after based on whether the upstream transcripts are \n",
    "##transcribed\n",
    "counts_bidirs_nooverlaps <- counts_bidirs_pos_neg[!counts_bidirs_pos_neg$bidirs\n",
    "                                                 %in%\n",
    "                                                  unique(overlaps_non_multiple$bidirs),]\n",
    "nrow(counts_bidirs_nooverlaps)\n",
    "head(counts_bidirs_nooverlaps, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "589176"
      ],
      "text/latex": [
       "589176"
      ],
      "text/markdown": [
       "589176"
      ],
      "text/plain": [
       "[1] 589176"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(c(overlaps_non_multiple$bidirs, as.character(overlaps_bidirs_multiple$Var1))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "529446"
      ],
      "text/latex": [
       "529446"
      ],
      "text/markdown": [
       "529446"
      ],
      "text/plain": [
       "[1] 529446"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "length(unique(overlaps_non_multiple$bidirs))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine bidirection pos, neg and both strand counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "847521"
      ],
      "text/latex": [
       "847521"
      ],
      "text/markdown": [
       "847521"
      ],
      "text/plain": [
       "[1] 847521"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>bidirs</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td>12182</td><td>12456</td><td>chr1:12182-12456</td><td> 1</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td> 0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>chr1</td><td>13264</td><td>13506</td><td>chr1:13264-13506</td><td>14</td><td>.</td><td>4</td><td>3</td><td>8</td><td>6</td><td>⋯</td><td>11</td><td>9</td><td>5</td><td>4</td><td>2</td><td>2</td><td>4</td><td>2</td><td>0</td><td>6</td></tr>\n",
       "\t<tr><td>chr1</td><td>14655</td><td>14811</td><td>chr1:14655-14811</td><td>14</td><td>.</td><td>1</td><td>0</td><td>0</td><td>1</td><td>⋯</td><td> 1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 1651\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " chrom & start & end & bidirs & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 & 12182 & 12456 & chr1:12182-12456 &  1 & . & 0 & 0 & 0 & 0 & ⋯ &  0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t chr1 & 13264 & 13506 & chr1:13264-13506 & 14 & . & 4 & 3 & 8 & 6 & ⋯ & 11 & 9 & 5 & 4 & 2 & 2 & 4 & 2 & 0 & 6\\\\\n",
       "\t chr1 & 14655 & 14811 & chr1:14655-14811 & 14 & . & 1 & 0 & 0 & 1 & ⋯ &  1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 1651\n",
       "\n",
       "| chrom &lt;chr&gt; | start &lt;int&gt; | end &lt;int&gt; | bidirs &lt;chr&gt; | score &lt;int&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr1 | 12182 | 12456 | chr1:12182-12456 |  1 | . | 0 | 0 | 0 | 0 | ⋯ |  0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| chr1 | 13264 | 13506 | chr1:13264-13506 | 14 | . | 4 | 3 | 8 | 6 | ⋯ | 11 | 9 | 5 | 4 | 2 | 2 | 4 | 2 | 0 | 6 |\n",
       "| chr1 | 14655 | 14811 | chr1:14655-14811 | 14 | . | 1 | 0 | 0 | 1 | ⋯ |  1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start end   bidirs           score strand SRR7266931 SRR7266932\n",
       "1 chr1  12182 12456 chr1:12182-12456  1    .      0          0         \n",
       "2 chr1  13264 13506 chr1:13264-13506 14    .      4          3         \n",
       "3 chr1  14655 14811 chr1:14655-14811 14    .      1          0         \n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1 0          0          ⋯  0         0          0          0         \n",
       "2 8          6          ⋯ 11         9          5          4         \n",
       "3 0          1          ⋯  1         0          0          0         \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0          0          0          0          0          0         \n",
       "2 2          2          4          2          0          6         \n",
       "3 0          0          0          0          0          0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts_bidirs <- rbind(counts_bidirs_nooverlaps,\n",
    "                       counts_pos_bidirs_overlaps, \n",
    "                       counts_neg_bidirs_overlaps)\n",
    "nrow(counts_bidirs)\n",
    "head(counts_bidirs, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Combine all counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>876410</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 876410\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 876410\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 876410   1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>43</td><td>4</td><td>45</td><td>34</td><td>⋯</td><td>12</td><td>10</td><td>38</td><td>30</td><td>42</td><td>62</td><td>53</td><td>44</td><td>49</td><td>61</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td> 0</td><td>0</td><td> 0</td><td> 0</td><td>⋯</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 1651\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2   & . & + &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1    & . & - & 43 & 4 & 45 & 34 & ⋯ & 12 & 10 & 38 & 30 & 42 & 62 & 53 & 44 & 49 & 61\\\\\n",
       "\t3 & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - &  0 & 0 &  0 &  0 & ⋯ &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0 &  0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 1651\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2   | . | + |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1    | . | - | 43 | 4 | 45 | 34 | ⋯ | 12 | 10 | 38 | 30 | 42 | 62 | 53 | 44 | 49 | 61 |\n",
       "| 3 | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - |  0 | 0 |  0 |  0 | ⋯ |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |  0 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript       score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2   .     +       0         0         \n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1    .     -      43         4         \n",
       "3 chr1  17368 17436 MIR6859-1:NR_106918.1 .     -       0         0         \n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1  0          0         ⋯  0          0          0          0        \n",
       "2 45         34         ⋯ 12         10         38         30        \n",
       "3  0          0         ⋯  0          0          0          0        \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1  0          0          0          0          0          0        \n",
       "2 42         62         53         44         49         61        \n",
       "3  0          0          0          0          0          0        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>876408</th><td>chrY</td><td>25633930</td><td>25634690</td><td>chrY:25633930-25634690</td><td>1</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>876409</th><td>chrY</td><td>25639750</td><td>25640170</td><td>chrY:25639750-25640170</td><td>1</td><td>.</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>876410</th><td>chrY</td><td>25738260</td><td>25738790</td><td>chrY:25738260-25738790</td><td>1</td><td>.</td><td>1</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 1651\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t876408 & chrY & 25633930 & 25634690 & chrY:25633930-25634690 & 1 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t876409 & chrY & 25639750 & 25640170 & chrY:25639750-25640170 & 1 & . & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\t876410 & chrY & 25738260 & 25738790 & chrY:25738260-25738790 & 1 & . & 1 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 1651\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;int&gt; | SRR7266932 &lt;int&gt; | SRR7266933 &lt;int&gt; | SRR7266934 &lt;int&gt; | ⋯ ⋯ | SRR3713716 &lt;int&gt; | SRR3713717 &lt;int&gt; | SRR3713718 &lt;int&gt; | SRR3713719 &lt;int&gt; | SRR8483105 &lt;int&gt; | SRR8483106 &lt;int&gt; | SRR8483107 &lt;int&gt; | SRR8483108 &lt;int&gt; | SRR8483109 &lt;int&gt; | SRR8483110 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 876408 | chrY | 25633930 | 25634690 | chrY:25633930-25634690 | 1 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 876409 | chrY | 25639750 | 25640170 | chrY:25639750-25640170 | 1 | . | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| 876410 | chrY | 25738260 | 25738790 | chrY:25738260-25738790 | 1 | . | 1 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "       chrom start    stop     gene_transcript        score strand SRR7266931\n",
       "876408 chrY  25633930 25634690 chrY:25633930-25634690 1     .      0         \n",
       "876409 chrY  25639750 25640170 chrY:25639750-25640170 1     .      0         \n",
       "876410 chrY  25738260 25738790 chrY:25738260-25738790 1     .      1         \n",
       "       SRR7266932 SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718\n",
       "876408 0          0          0          ⋯ 0          0          0         \n",
       "876409 0          0          0          ⋯ 0          0          0         \n",
       "876410 0          0          0          ⋯ 0          0          0         \n",
       "       SRR3713719 SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109\n",
       "876408 0          0          0          0          0          0         \n",
       "876409 0          0          0          0          0          0         \n",
       "876410 0          0          0          0          0          0         \n",
       "       SRR8483110\n",
       "876408 0         \n",
       "876409 0         \n",
       "876410 0         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "counts_bidir_genes_trunc <- as.data.frame(rbind(counts_genes, \n",
    "                                          counts_bidirs,\n",
    "                                        use.names=FALSE))\n",
    "\n",
    "dim(counts_bidir_genes_trunc)\n",
    "head(counts_bidir_genes_trunc, 3)\n",
    "tail(counts_bidir_genes_trunc, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalize counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>876410</li><li>1651</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 876410\n",
       "\\item 1651\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 876410\n",
       "2. 1651\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 876410   1651"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 1651</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR3713719</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.1448111</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>chr1</td><td>30365</td><td>30503</td><td>MIR1302-2:NR_036051.1</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>chr1</td><td>34610</td><td>35591</td><td>FAM138A:NR_026818.1  </td><td>.</td><td>-</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 1651\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713716 & SRR3713717 & SRR3713718 & SRR3713719 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2   & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1    & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.05880973 & 0.06709094 & 0.2776161 & 0.1448111 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t3 & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t4 & chr1 & 30365 & 30503 & MIR1302-2:NR\\_036051.1 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t5 & chr1 & 34610 & 35591 & FAM138A:NR\\_026818.1   & . & - & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 1651\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR3713719 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2   | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1    | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.05880973 | 0.06709094 | 0.2776161 | 0.1448111 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 3 | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 4 | chr1 | 30365 | 30503 | MIR1302-2:NR_036051.1 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 5 | chr1 | 34610 | 35591 | FAM138A:NR_026818.1   | . | - | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript       score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2   .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1    .     -      0.2372728  0.04812349\n",
       "3 chr1  17368 17436 MIR6859-1:NR_106918.1 .     -      0.0000000  0.00000000\n",
       "4 chr1  30365 30503 MIR1302-2:NR_036051.1 .     +      0.0000000  0.00000000\n",
       "5 chr1  34610 35591 FAM138A:NR_026818.1   .     -      0.0000000  0.00000000\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2  .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713716 SRR3713717 SRR3713718 SRR3713719\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.0000000  0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.05880973 0.06709094 0.2776161  0.1448111 \n",
       "3 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.0000000  0.0000000 \n",
       "4 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.0000000  0.0000000 \n",
       "5 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.0000000  0.0000000 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.0000000  0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "3 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "4 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "5 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##get variables for the TPM calculation function\n",
    "##and calculate normalized counts\n",
    "\n",
    "#1. transcript lenths\n",
    "transcript_lengths <- (counts_bidir_genes_trunc$stop - counts_bidir_genes_trunc$start) +1\n",
    "\n",
    "#2. number of columns in the dataframes\n",
    "ncolumns <- ncol(counts_bidir_genes_trunc)\n",
    "\n",
    "#3. get normalized counts by sample\n",
    "gene_bidir_tpms <- tpm_calculation(counts_bidir_genes_trunc[,7:ncolumns], \n",
    "                        transcript_lengths)\n",
    "\n",
    "#4. merge the transcript annotations with normalized counts data.frame\n",
    "gene_bidir_tpm_df <- cbind(counts_bidir_genes_trunc[,1:6],\n",
    "                           as.data.frame(gene_bidir_tpms))\n",
    "\n",
    "#summary of data.frame\n",
    "dim(gene_bidir_tpm_df)\n",
    "head(gene_bidir_tpm_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save normalized counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_df,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/gene_bidir_tpm.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_bidir_tpm_qc <- gene_bidir_tpm_df[,colnames(gene_bidir_tpm_df) %in%\n",
    "                                         c('chrom','start','stop',\n",
    "                                            'gene_transcript','score',\n",
    "                                            'strand',metadata_celltype$sample_name),\n",
    "                                           with=FALSE]\n",
    "dim(gene_bidir_tpm_qc)\n",
    "head(gene_bidir_tpm_qc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#binary for trancribed or not\n",
    "tpm_matrix_binary <- ifelse(gene_bidir_tpm_qc[,c(7:757)] != 0, 1, 0)\n",
    "head(tpm_matrix_binary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_bidir_tpm_keep2 <- gene_bidir_tpm_qc[rowSums(tpm_matrix_binary)>2,]\n",
    "dim(gene_bidir_tpm_keep2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_keep2,\n",
    "                   \"/scratch/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/counts/normalized_counts/gene_bidir_tpm_qc.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>876410</li><li>757</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 876410\n",
       "\\item 757\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 876410\n",
       "2. 757\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 876410    757"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2   & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1    & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t3 & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2   | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1    | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 3 | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript       score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2   .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1    .     -      0.2372728  0.04812349\n",
       "3 chr1  17368 17436 MIR6859-1:NR_106918.1 .     -      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "3 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "3 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##filter all bidirectional and gene transcripts\n",
    "gene_bidir_tpm_samples_filtered <- cbind(gene_bidir_tpm_df[,1:6], \n",
    "                                         gene_bidir_tpm_df[,\n",
    "                                                           colnames(gene_bidir_tpm_df)\n",
    "                                                           %in% filtered_samples$sample_name])\n",
    "                                                           #with=FALSE])\n",
    "dim(gene_bidir_tpm_samples_filtered)\n",
    "head(gene_bidir_tpm_samples_filtered, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get summary statistics for counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>876410</li><li>18</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 876410\n",
       "\\item 18\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 876410\n",
       "2. 18\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 876410     18"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 18</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>length</th><th scope=col>stdev</th><th scope=col>variance</th><th scope=col>sum</th><th scope=col>mean</th><th scope=col>median</th><th scope=col>coefvar</th><th scope=col>max</th><th scope=col>min</th><th scope=col>num_transcribed</th><th scope=col>percent_transcribed</th><th scope=col>transcript_type</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2  </td><td>.</td><td>+</td><td> 1787</td><td>0.07496376</td><td>0.005619565</td><td>  7.345779</td><td>0.00978133</td><td>0.0000000</td><td>7.6639642</td><td>1.256177</td><td>0</td><td> 29</td><td> 3.861518</td><td>genes</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1   </td><td>.</td><td>-</td><td>14260</td><td>0.20321583</td><td>0.041296674</td><td>186.403807</td><td>0.24820747</td><td>0.1942185</td><td>0.8187338</td><td>1.417130</td><td>0</td><td>749</td><td>99.733688</td><td>genes</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>chr1</td><td>17368</td><td>17436</td><td>MIR6859-1:NR_106918.1</td><td>.</td><td>-</td><td>   69</td><td>0.00000000</td><td>0.000000000</td><td>  0.000000</td><td>0.00000000</td><td>0.0000000</td><td>      NaN</td><td>0.000000</td><td>0</td><td>  0</td><td> 0.000000</td><td>genes</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 18\n",
       "\\begin{tabular}{r|llllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & length & stdev & variance & sum & mean & median & coefvar & max & min & num\\_transcribed & percent\\_transcribed & transcript\\_type\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2   & . & + &  1787 & 0.07496376 & 0.005619565 &   7.345779 & 0.00978133 & 0.0000000 & 7.6639642 & 1.256177 & 0 &  29 &  3.861518 & genes\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1    & . & - & 14260 & 0.20321583 & 0.041296674 & 186.403807 & 0.24820747 & 0.1942185 & 0.8187338 & 1.417130 & 0 & 749 & 99.733688 & genes\\\\\n",
       "\t3 & chr1 & 17368 & 17436 & MIR6859-1:NR\\_106918.1 & . & - &    69 & 0.00000000 & 0.000000000 &   0.000000 & 0.00000000 & 0.0000000 &       NaN & 0.000000 & 0 &   0 &  0.000000 & genes\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 18\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | length &lt;dbl&gt; | stdev &lt;dbl&gt; | variance &lt;dbl&gt; | sum &lt;dbl&gt; | mean &lt;dbl&gt; | median &lt;dbl&gt; | coefvar &lt;dbl&gt; | max &lt;dbl&gt; | min &lt;dbl&gt; | num_transcribed &lt;dbl&gt; | percent_transcribed &lt;dbl&gt; | transcript_type &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2   | . | + |  1787 | 0.07496376 | 0.005619565 |   7.345779 | 0.00978133 | 0.0000000 | 7.6639642 | 1.256177 | 0 |  29 |  3.861518 | genes |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1    | . | - | 14260 | 0.20321583 | 0.041296674 | 186.403807 | 0.24820747 | 0.1942185 | 0.8187338 | 1.417130 | 0 | 749 | 99.733688 | genes |\n",
       "| 3 | chr1 | 17368 | 17436 | MIR6859-1:NR_106918.1 | . | - |    69 | 0.00000000 | 0.000000000 |   0.000000 | 0.00000000 | 0.0000000 |       NaN | 0.000000 | 0 |   0 |  0.000000 | genes |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript       score strand length stdev     \n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2   .     +       1787  0.07496376\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1    .     -      14260  0.20321583\n",
       "3 chr1  17368 17436 MIR6859-1:NR_106918.1 .     -         69  0.00000000\n",
       "  variance    sum        mean       median    coefvar   max      min\n",
       "1 0.005619565   7.345779 0.00978133 0.0000000 7.6639642 1.256177 0  \n",
       "2 0.041296674 186.403807 0.24820747 0.1942185 0.8187338 1.417130 0  \n",
       "3 0.000000000   0.000000 0.00000000 0.0000000       NaN 0.000000 0  \n",
       "  num_transcribed percent_transcribed transcript_type\n",
       "1  29              3.861518           genes          \n",
       "2 749             99.733688           genes          \n",
       "3   0              0.000000           genes          "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#convert dt to df\n",
    "gene_bidir_tpm_samples_filtered_df <- as.data.frame(gene_bidir_tpm_samples_filtered) ##gene_bidir_tpm_df\n",
    "\n",
    "genes <- gene_bidir_tpm_df[!gene_bidir_tpm_df$gene_transcript \n",
    "                                         %in%\n",
    "                                         counts_bidirs$bidirs,]\n",
    "                                         \n",
    "##get summary statistics based on the filtered samples\n",
    "gene_bidir_tpm_filtered_summary <- get_summary_statistics(gene_bidir_tpm_samples_filtered_df,\n",
    "                                                genes$gene_transcript,\n",
    "                                                counts_bidirs$bidirs)\n",
    "dim(gene_bidir_tpm_filtered_summary)\n",
    "head(gene_bidir_tpm_filtered_summary, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "##add a column with bidirectional types (intergenic, intragenic, intragenic and overlapping pos+neg genes)\n",
    "gene_bidir_tpm_filtered_summary$bidirectional_type <- ifelse(gene_bidir_tpm_filtered_summary$gene_transcript %in% counts_genes$gene_transcript,\n",
    "                                                 \"genes\",\n",
    "                                                 ifelse(gene_bidir_tpm_filtered_summary$gene_transcript %in% bidir_no_overlaps$bidirs,\n",
    "                                                        \"intergenic_bidirectional\",\n",
    "                                                       ifelse(gene_bidir_tpm_filtered_summary$gene_transcript %in% as.character(overlaps_bidirs_non_multiple$Var1),\n",
    "                                                             \"intragenic_bidirectional\",\n",
    "                                                             ifelse(gene_bidir_tpm_filtered_summary$gene_transcript %in% as.character(overlaps_bidirs_multiple$Var1),\n",
    "                                                                   \"intragenic_bidirectional_pos_neg\",\"NA\"))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "                           genes         intergenic_bidirectional \n",
       "                           28889                           258345 \n",
       "        intragenic_bidirectional intragenic_bidirectional_pos_neg \n",
       "                          529446                            59730 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(gene_bidir_tpm_filtered_summary$bidirectional_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_filtered_summary,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/summary_gene_bidir_tpm_filtered.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter transcripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Remove transcripts with 0 total counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "837180"
      ],
      "text/latex": [
       "837180"
      ],
      "text/markdown": [
       "837180"
      ],
      "text/plain": [
       "[1] 837180"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_bidir_tpm_filtered_summary_txped <- subset(gene_bidir_tpm_filtered_summary, \n",
    "                                                num_transcribed >=10)\n",
    "nrow(gene_bidir_tpm_filtered_summary_txped)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "837180"
      ],
      "text/latex": [
       "837180"
      ],
      "text/markdown": [
       "837180"
      ],
      "text/plain": [
       "[1] 837180"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1  </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1   & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1   | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript      score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2  .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1   .     -      0.2372728  0.04812349\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2 .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##get transcripts with counts in 10 or more samples\n",
    "gene_bidir_tpm_samples_filtered_txped <- gene_bidir_tpm_samples_filtered[gene_bidir_tpm_samples_filtered$gene_transcript \n",
    "                                                                        %in%\n",
    "                                                                        gene_bidir_tpm_filtered_summary_txped$gene_transcript,]\n",
    "nrow(gene_bidir_tpm_samples_filtered_txped)\n",
    "head(gene_bidir_tpm_samples_filtered_txped, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Split bidirectionals for easy processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intergenic bidirectionals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "287234"
      ],
      "text/latex": [
       "287234"
      ],
      "text/markdown": [
       "287234"
      ],
      "text/plain": [
       "[1] 287234"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intergenic_bidirs_genes <- c(genes$gene_transcript,\n",
    "                             bidir_no_overlaps$bidirs)\n",
    "length(intergenic_bidirs_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>280180</li><li>757</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 280180\n",
       "\\item 757\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 280180\n",
       "2. 757\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 280180    757"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1  </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1   & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1   | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript      score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2  .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1   .     -      0.2372728  0.04812349\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2 .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_bidir_tpm_intergenic <- gene_bidir_tpm_samples_filtered_txped[gene_bidir_tpm_samples_filtered_txped$gene_transcript \n",
    "                                                                          %in%\n",
    "                                                                   intergenic_bidirs_genes,]\n",
    "dim(gene_bidir_tpm_intergenic)\n",
    "head(gene_bidir_tpm_intergenic, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "bidirectional         genes \n",
       "       252181         27999 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#how many bidirectionals and genes are in the set\n",
    "check_intergenic <- gene_bidir_tpm_filtered_summary[gene_bidir_tpm_filtered_summary$gene_transcript \n",
    "                                 %in% \n",
    "                                 gene_bidir_tpm_intergenic$gene_transcript,]\n",
    "\n",
    "table(check_intergenic$transcript_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_intergenic,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/gene_bidir_tpm_intergenic.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Intragenic bidirectionals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "589176"
      ],
      "text/latex": [
       "589176"
      ],
      "text/markdown": [
       "589176"
      ],
      "text/plain": [
       "[1] 589176"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##list of all intergenic bidirectionals\n",
    "set.seed(42)\n",
    "\n",
    "#shuffle the list of bidirectionals so that when they are split into chunks,\n",
    "#there is roughly an even split of chromosomes\n",
    "bidir_overlaps_list <- sample(unique(bidir_overlaps$bidirs))\n",
    "length(bidir_overlaps_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "196392"
      ],
      "text/latex": [
       "196392"
      ],
      "text/markdown": [
       "196392"
      ],
      "text/plain": [
       "[1] 196392"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#divide these transcripts into two sets\n",
    "length(bidir_overlaps_list)/3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "225281"
      ],
      "text/latex": [
       "225281"
      ],
      "text/markdown": [
       "225281"
      ],
      "text/plain": [
       "[1] 225281"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intragenic_bidirs_genes_A <- c(genes$gene_transcript,\n",
    "                             head(bidir_overlaps_list, 196392))\n",
    "length(intragenic_bidirs_genes_A)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "225281"
      ],
      "text/latex": [
       "225281"
      ],
      "text/markdown": [
       "225281"
      ],
      "text/plain": [
       "[1] 225281"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intragenic_bidirs_genes_B <- c(genes$gene_transcript,\n",
    "                             tail(bidir_overlaps_list, 196392))\n",
    "length(intragenic_bidirs_genes_B)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "225281"
      ],
      "text/latex": [
       "225281"
      ],
      "text/markdown": [
       "225281"
      ],
      "text/plain": [
       "[1] 225281"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "intragenic_bidirs_genes_C <- c(genes$gene_transcript, \n",
    "                               setdiff(bidir_overlaps_list,\n",
    "                                         c(intragenic_bidirs_genes_A,intragenic_bidirs_genes_B)))\n",
    "length(intragenic_bidirs_genes_C)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'chr10:67151630-67151942'"
      ],
      "text/latex": [
       "'chr10:67151630-67151942'"
      ],
      "text/markdown": [
       "'chr10:67151630-67151942'"
      ],
      "text/plain": [
       "[1] \"chr10:67151630-67151942\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Check that the bidirectional transcripts do not overlap\n",
    "intersect(tail(bidir_overlaps_list, 294588),\n",
    "         head(bidir_overlaps_list, 294589))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>213784</li><li>757</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 213784\n",
       "\\item 757\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 213784\n",
       "2. 757\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 213784    757"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1  </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1   & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1   | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript      score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2  .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1   .     -      0.2372728  0.04812349\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2 .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_bidir_tpm_intragenic_A <- gene_bidir_tpm_samples_filtered_txped[gene_bidir_tpm_samples_filtered_txped$gene_transcript \n",
    "                                                                          %in%\n",
    "                                                                   intragenic_bidirs_genes_A,]\n",
    "dim(gene_bidir_tpm_intragenic_A)\n",
    "head(gene_bidir_tpm_intragenic_A, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>213646</li><li>757</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 213646\n",
       "\\item 757\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 213646\n",
       "2. 757\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 213646    757"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1  </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1   & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1   | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript      score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2  .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1   .     -      0.2372728  0.04812349\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2 .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_bidir_tpm_intragenic_B <- gene_bidir_tpm_samples_filtered_txped[gene_bidir_tpm_samples_filtered_txped$gene_transcript \n",
    "                                                                          %in%\n",
    "                                                                   intragenic_bidirs_genes_B,]\n",
    "dim(gene_bidir_tpm_intragenic_B)\n",
    "head(gene_bidir_tpm_intragenic_B, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>213567</li><li>757</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 213567\n",
       "\\item 757\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 213567\n",
       "2. 757\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 213567    757"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 3 × 757</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>chrom</th><th scope=col>start</th><th scope=col>stop</th><th scope=col>gene_transcript</th><th scope=col>score</th><th scope=col>strand</th><th scope=col>SRR7266931</th><th scope=col>SRR7266932</th><th scope=col>SRR7266933</th><th scope=col>SRR7266934</th><th scope=col>⋯</th><th scope=col>SRR3713715</th><th scope=col>SRR3713716</th><th scope=col>SRR3713717</th><th scope=col>SRR3713718</th><th scope=col>SRR8483105</th><th scope=col>SRR8483106</th><th scope=col>SRR8483107</th><th scope=col>SRR8483108</th><th scope=col>SRR8483109</th><th scope=col>SRR8483110</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>⋯</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>chr1</td><td>12623</td><td>14409</td><td>DDX11L1:NR_046018.2 </td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>chr1</td><td>14361</td><td>28620</td><td>WASH7P:NR_024540.1  </td><td>.</td><td>-</td><td>0.2372728</td><td>0.04812349</td><td>0.247163</td><td>0.180835</td><td>⋯</td><td>0.09228832</td><td>0.05880973</td><td>0.06709094</td><td>0.2776161</td><td>0.3800877</td><td>0.4256579</td><td>0.3585765</td><td>0.3028641</td><td>0.3857214</td><td>0.4399284</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>chr1</td><td>66168</td><td>71585</td><td>OR4F5:NM_001005484.2</td><td>.</td><td>+</td><td>0.0000000</td><td>0.00000000</td><td>0.000000</td><td>0.000000</td><td>⋯</td><td>0.00000000</td><td>0.00000000</td><td>0.00000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td><td>0.0000000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 3 × 757\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & chrom & start & stop & gene\\_transcript & score & strand & SRR7266931 & SRR7266932 & SRR7266933 & SRR7266934 & ⋯ & SRR3713715 & SRR3713716 & SRR3713717 & SRR3713718 & SRR8483105 & SRR8483106 & SRR8483107 & SRR8483108 & SRR8483109 & SRR8483110\\\\\n",
       "  & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & ⋯ & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & chr1 & 12623 & 14409 & DDX11L1:NR\\_046018.2  & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\t2 & chr1 & 14361 & 28620 & WASH7P:NR\\_024540.1   & . & - & 0.2372728 & 0.04812349 & 0.247163 & 0.180835 & ⋯ & 0.09228832 & 0.05880973 & 0.06709094 & 0.2776161 & 0.3800877 & 0.4256579 & 0.3585765 & 0.3028641 & 0.3857214 & 0.4399284\\\\\n",
       "\t6 & chr1 & 66168 & 71585 & OR4F5:NM\\_001005484.2 & . & + & 0.0000000 & 0.00000000 & 0.000000 & 0.000000 & ⋯ & 0.00000000 & 0.00000000 & 0.00000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000 & 0.0000000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 3 × 757\n",
       "\n",
       "| <!--/--> | chrom &lt;chr&gt; | start &lt;int&gt; | stop &lt;int&gt; | gene_transcript &lt;chr&gt; | score &lt;chr&gt; | strand &lt;chr&gt; | SRR7266931 &lt;dbl&gt; | SRR7266932 &lt;dbl&gt; | SRR7266933 &lt;dbl&gt; | SRR7266934 &lt;dbl&gt; | ⋯ ⋯ | SRR3713715 &lt;dbl&gt; | SRR3713716 &lt;dbl&gt; | SRR3713717 &lt;dbl&gt; | SRR3713718 &lt;dbl&gt; | SRR8483105 &lt;dbl&gt; | SRR8483106 &lt;dbl&gt; | SRR8483107 &lt;dbl&gt; | SRR8483108 &lt;dbl&gt; | SRR8483109 &lt;dbl&gt; | SRR8483110 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | chr1 | 12623 | 14409 | DDX11L1:NR_046018.2  | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "| 2 | chr1 | 14361 | 28620 | WASH7P:NR_024540.1   | . | - | 0.2372728 | 0.04812349 | 0.247163 | 0.180835 | ⋯ | 0.09228832 | 0.05880973 | 0.06709094 | 0.2776161 | 0.3800877 | 0.4256579 | 0.3585765 | 0.3028641 | 0.3857214 | 0.4399284 |\n",
       "| 6 | chr1 | 66168 | 71585 | OR4F5:NM_001005484.2 | . | + | 0.0000000 | 0.00000000 | 0.000000 | 0.000000 | ⋯ | 0.00000000 | 0.00000000 | 0.00000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 | 0.0000000 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start stop  gene_transcript      score strand SRR7266931 SRR7266932\n",
       "1 chr1  12623 14409 DDX11L1:NR_046018.2  .     +      0.0000000  0.00000000\n",
       "2 chr1  14361 28620 WASH7P:NR_024540.1   .     -      0.2372728  0.04812349\n",
       "6 chr1  66168 71585 OR4F5:NM_001005484.2 .     +      0.0000000  0.00000000\n",
       "  SRR7266933 SRR7266934 ⋯ SRR3713715 SRR3713716 SRR3713717 SRR3713718\n",
       "1 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "2 0.247163   0.180835   ⋯ 0.09228832 0.05880973 0.06709094 0.2776161 \n",
       "6 0.000000   0.000000   ⋯ 0.00000000 0.00000000 0.00000000 0.0000000 \n",
       "  SRR8483105 SRR8483106 SRR8483107 SRR8483108 SRR8483109 SRR8483110\n",
       "1 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 \n",
       "2 0.3800877  0.4256579  0.3585765  0.3028641  0.3857214  0.4399284 \n",
       "6 0.0000000  0.0000000  0.0000000  0.0000000  0.0000000  0.0000000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_bidir_tpm_intragenic_C <- gene_bidir_tpm_samples_filtered_txped[gene_bidir_tpm_samples_filtered_txped$gene_transcript \n",
    "                                                                          %in%\n",
    "                                                                   intragenic_bidirs_genes_C,]\n",
    "dim(gene_bidir_tpm_intragenic_C)\n",
    "head(gene_bidir_tpm_intragenic_C, 3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_intragenic_A,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/gene_bidir_tpm_intragenic_A.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_intragenic_B,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/gene_bidir_tpm_intragenic_B.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.table::fwrite(gene_bidir_tpm_intragenic_C,\n",
    "                   \"/Users/rusi2317/projects/meta_analysis_qc_reviews/processed_data/normalized_counts/gene_bidir_tpm_intragenic_C.tsv.gz\",\n",
    "                  sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check per-chromosome number of transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 24 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>23247</td></tr>\n",
       "\t<tr><td>chr10</td><td>12333</td></tr>\n",
       "\t<tr><td>chr11</td><td>11868</td></tr>\n",
       "\t<tr><td>chr12</td><td>13062</td></tr>\n",
       "\t<tr><td>chr13</td><td> 9058</td></tr>\n",
       "\t<tr><td>chr14</td><td> 8964</td></tr>\n",
       "\t<tr><td>chr15</td><td> 8094</td></tr>\n",
       "\t<tr><td>chr16</td><td> 7696</td></tr>\n",
       "\t<tr><td>chr17</td><td> 7634</td></tr>\n",
       "\t<tr><td>chr18</td><td> 6611</td></tr>\n",
       "\t<tr><td>chr19</td><td> 5344</td></tr>\n",
       "\t<tr><td>chr2 </td><td>24311</td></tr>\n",
       "\t<tr><td>chr20</td><td> 6843</td></tr>\n",
       "\t<tr><td>chr21</td><td> 4182</td></tr>\n",
       "\t<tr><td>chr22</td><td> 3809</td></tr>\n",
       "\t<tr><td>chr3 </td><td>17343</td></tr>\n",
       "\t<tr><td>chr4 </td><td>17130</td></tr>\n",
       "\t<tr><td>chr5 </td><td>17935</td></tr>\n",
       "\t<tr><td>chr6 </td><td>17753</td></tr>\n",
       "\t<tr><td>chr7 </td><td>14590</td></tr>\n",
       "\t<tr><td>chr8 </td><td>14599</td></tr>\n",
       "\t<tr><td>chr9 </td><td>12786</td></tr>\n",
       "\t<tr><td>chrX </td><td>13355</td></tr>\n",
       "\t<tr><td>chrY </td><td> 1633</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Var1 & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 23247\\\\\n",
       "\t chr10 & 12333\\\\\n",
       "\t chr11 & 11868\\\\\n",
       "\t chr12 & 13062\\\\\n",
       "\t chr13 &  9058\\\\\n",
       "\t chr14 &  8964\\\\\n",
       "\t chr15 &  8094\\\\\n",
       "\t chr16 &  7696\\\\\n",
       "\t chr17 &  7634\\\\\n",
       "\t chr18 &  6611\\\\\n",
       "\t chr19 &  5344\\\\\n",
       "\t chr2  & 24311\\\\\n",
       "\t chr20 &  6843\\\\\n",
       "\t chr21 &  4182\\\\\n",
       "\t chr22 &  3809\\\\\n",
       "\t chr3  & 17343\\\\\n",
       "\t chr4  & 17130\\\\\n",
       "\t chr5  & 17935\\\\\n",
       "\t chr6  & 17753\\\\\n",
       "\t chr7  & 14590\\\\\n",
       "\t chr8  & 14599\\\\\n",
       "\t chr9  & 12786\\\\\n",
       "\t chrX  & 13355\\\\\n",
       "\t chrY  &  1633\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 2\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 23247 |\n",
       "| chr10 | 12333 |\n",
       "| chr11 | 11868 |\n",
       "| chr12 | 13062 |\n",
       "| chr13 |  9058 |\n",
       "| chr14 |  8964 |\n",
       "| chr15 |  8094 |\n",
       "| chr16 |  7696 |\n",
       "| chr17 |  7634 |\n",
       "| chr18 |  6611 |\n",
       "| chr19 |  5344 |\n",
       "| chr2  | 24311 |\n",
       "| chr20 |  6843 |\n",
       "| chr21 |  4182 |\n",
       "| chr22 |  3809 |\n",
       "| chr3  | 17343 |\n",
       "| chr4  | 17130 |\n",
       "| chr5  | 17935 |\n",
       "| chr6  | 17753 |\n",
       "| chr7  | 14590 |\n",
       "| chr8  | 14599 |\n",
       "| chr9  | 12786 |\n",
       "| chrX  | 13355 |\n",
       "| chrY  |  1633 |\n",
       "\n"
      ],
      "text/plain": [
       "   Var1  Freq \n",
       "1  chr1  23247\n",
       "2  chr10 12333\n",
       "3  chr11 11868\n",
       "4  chr12 13062\n",
       "5  chr13  9058\n",
       "6  chr14  8964\n",
       "7  chr15  8094\n",
       "8  chr16  7696\n",
       "9  chr17  7634\n",
       "10 chr18  6611\n",
       "11 chr19  5344\n",
       "12 chr2  24311\n",
       "13 chr20  6843\n",
       "14 chr21  4182\n",
       "15 chr22  3809\n",
       "16 chr3  17343\n",
       "17 chr4  17130\n",
       "18 chr5  17935\n",
       "19 chr6  17753\n",
       "20 chr7  14590\n",
       "21 chr8  14599\n",
       "22 chr9  12786\n",
       "23 chrX  13355\n",
       "24 chrY   1633"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.data.frame(table(gene_bidir_tpm_intergenic$chrom))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 24 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>20462</td></tr>\n",
       "\t<tr><td>chr10</td><td>10011</td></tr>\n",
       "\t<tr><td>chr11</td><td>10635</td></tr>\n",
       "\t<tr><td>chr12</td><td>10961</td></tr>\n",
       "\t<tr><td>chr13</td><td> 5071</td></tr>\n",
       "\t<tr><td>chr14</td><td> 6800</td></tr>\n",
       "\t<tr><td>chr15</td><td> 7458</td></tr>\n",
       "\t<tr><td>chr16</td><td> 7505</td></tr>\n",
       "\t<tr><td>chr17</td><td>10301</td></tr>\n",
       "\t<tr><td>chr18</td><td> 4160</td></tr>\n",
       "\t<tr><td>chr19</td><td> 8875</td></tr>\n",
       "\t<tr><td>chr2 </td><td>16214</td></tr>\n",
       "\t<tr><td>chr20</td><td> 5393</td></tr>\n",
       "\t<tr><td>chr21</td><td> 2687</td></tr>\n",
       "\t<tr><td>chr22</td><td> 4429</td></tr>\n",
       "\t<tr><td>chr3 </td><td>13856</td></tr>\n",
       "\t<tr><td>chr4 </td><td> 9922</td></tr>\n",
       "\t<tr><td>chr5 </td><td>11643</td></tr>\n",
       "\t<tr><td>chr6 </td><td>11316</td></tr>\n",
       "\t<tr><td>chr7 </td><td>11376</td></tr>\n",
       "\t<tr><td>chr8 </td><td> 9092</td></tr>\n",
       "\t<tr><td>chr9 </td><td> 8738</td></tr>\n",
       "\t<tr><td>chrX </td><td> 6685</td></tr>\n",
       "\t<tr><td>chrY </td><td>  194</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Var1 & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 20462\\\\\n",
       "\t chr10 & 10011\\\\\n",
       "\t chr11 & 10635\\\\\n",
       "\t chr12 & 10961\\\\\n",
       "\t chr13 &  5071\\\\\n",
       "\t chr14 &  6800\\\\\n",
       "\t chr15 &  7458\\\\\n",
       "\t chr16 &  7505\\\\\n",
       "\t chr17 & 10301\\\\\n",
       "\t chr18 &  4160\\\\\n",
       "\t chr19 &  8875\\\\\n",
       "\t chr2  & 16214\\\\\n",
       "\t chr20 &  5393\\\\\n",
       "\t chr21 &  2687\\\\\n",
       "\t chr22 &  4429\\\\\n",
       "\t chr3  & 13856\\\\\n",
       "\t chr4  &  9922\\\\\n",
       "\t chr5  & 11643\\\\\n",
       "\t chr6  & 11316\\\\\n",
       "\t chr7  & 11376\\\\\n",
       "\t chr8  &  9092\\\\\n",
       "\t chr9  &  8738\\\\\n",
       "\t chrX  &  6685\\\\\n",
       "\t chrY  &   194\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 2\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 20462 |\n",
       "| chr10 | 10011 |\n",
       "| chr11 | 10635 |\n",
       "| chr12 | 10961 |\n",
       "| chr13 |  5071 |\n",
       "| chr14 |  6800 |\n",
       "| chr15 |  7458 |\n",
       "| chr16 |  7505 |\n",
       "| chr17 | 10301 |\n",
       "| chr18 |  4160 |\n",
       "| chr19 |  8875 |\n",
       "| chr2  | 16214 |\n",
       "| chr20 |  5393 |\n",
       "| chr21 |  2687 |\n",
       "| chr22 |  4429 |\n",
       "| chr3  | 13856 |\n",
       "| chr4  |  9922 |\n",
       "| chr5  | 11643 |\n",
       "| chr6  | 11316 |\n",
       "| chr7  | 11376 |\n",
       "| chr8  |  9092 |\n",
       "| chr9  |  8738 |\n",
       "| chrX  |  6685 |\n",
       "| chrY  |   194 |\n",
       "\n"
      ],
      "text/plain": [
       "   Var1  Freq \n",
       "1  chr1  20462\n",
       "2  chr10 10011\n",
       "3  chr11 10635\n",
       "4  chr12 10961\n",
       "5  chr13  5071\n",
       "6  chr14  6800\n",
       "7  chr15  7458\n",
       "8  chr16  7505\n",
       "9  chr17 10301\n",
       "10 chr18  4160\n",
       "11 chr19  8875\n",
       "12 chr2  16214\n",
       "13 chr20  5393\n",
       "14 chr21  2687\n",
       "15 chr22  4429\n",
       "16 chr3  13856\n",
       "17 chr4   9922\n",
       "18 chr5  11643\n",
       "19 chr6  11316\n",
       "20 chr7  11376\n",
       "21 chr8   9092\n",
       "22 chr9   8738\n",
       "23 chrX   6685\n",
       "24 chrY    194"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.data.frame(table(gene_bidir_tpm_intragenic_A$chrom))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 24 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>20635</td></tr>\n",
       "\t<tr><td>chr10</td><td> 9951</td></tr>\n",
       "\t<tr><td>chr11</td><td>10892</td></tr>\n",
       "\t<tr><td>chr12</td><td>10824</td></tr>\n",
       "\t<tr><td>chr13</td><td> 5165</td></tr>\n",
       "\t<tr><td>chr14</td><td> 6768</td></tr>\n",
       "\t<tr><td>chr15</td><td> 7403</td></tr>\n",
       "\t<tr><td>chr16</td><td> 7514</td></tr>\n",
       "\t<tr><td>chr17</td><td>10333</td></tr>\n",
       "\t<tr><td>chr18</td><td> 4199</td></tr>\n",
       "\t<tr><td>chr19</td><td> 8918</td></tr>\n",
       "\t<tr><td>chr2 </td><td>16139</td></tr>\n",
       "\t<tr><td>chr20</td><td> 5466</td></tr>\n",
       "\t<tr><td>chr21</td><td> 2654</td></tr>\n",
       "\t<tr><td>chr22</td><td> 4405</td></tr>\n",
       "\t<tr><td>chr3 </td><td>13532</td></tr>\n",
       "\t<tr><td>chr4 </td><td> 9908</td></tr>\n",
       "\t<tr><td>chr5 </td><td>11491</td></tr>\n",
       "\t<tr><td>chr6 </td><td>11406</td></tr>\n",
       "\t<tr><td>chr7 </td><td>11618</td></tr>\n",
       "\t<tr><td>chr8 </td><td> 8895</td></tr>\n",
       "\t<tr><td>chr9 </td><td> 8590</td></tr>\n",
       "\t<tr><td>chrX </td><td> 6772</td></tr>\n",
       "\t<tr><td>chrY </td><td>  168</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Var1 & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 20635\\\\\n",
       "\t chr10 &  9951\\\\\n",
       "\t chr11 & 10892\\\\\n",
       "\t chr12 & 10824\\\\\n",
       "\t chr13 &  5165\\\\\n",
       "\t chr14 &  6768\\\\\n",
       "\t chr15 &  7403\\\\\n",
       "\t chr16 &  7514\\\\\n",
       "\t chr17 & 10333\\\\\n",
       "\t chr18 &  4199\\\\\n",
       "\t chr19 &  8918\\\\\n",
       "\t chr2  & 16139\\\\\n",
       "\t chr20 &  5466\\\\\n",
       "\t chr21 &  2654\\\\\n",
       "\t chr22 &  4405\\\\\n",
       "\t chr3  & 13532\\\\\n",
       "\t chr4  &  9908\\\\\n",
       "\t chr5  & 11491\\\\\n",
       "\t chr6  & 11406\\\\\n",
       "\t chr7  & 11618\\\\\n",
       "\t chr8  &  8895\\\\\n",
       "\t chr9  &  8590\\\\\n",
       "\t chrX  &  6772\\\\\n",
       "\t chrY  &   168\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 2\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 20635 |\n",
       "| chr10 |  9951 |\n",
       "| chr11 | 10892 |\n",
       "| chr12 | 10824 |\n",
       "| chr13 |  5165 |\n",
       "| chr14 |  6768 |\n",
       "| chr15 |  7403 |\n",
       "| chr16 |  7514 |\n",
       "| chr17 | 10333 |\n",
       "| chr18 |  4199 |\n",
       "| chr19 |  8918 |\n",
       "| chr2  | 16139 |\n",
       "| chr20 |  5466 |\n",
       "| chr21 |  2654 |\n",
       "| chr22 |  4405 |\n",
       "| chr3  | 13532 |\n",
       "| chr4  |  9908 |\n",
       "| chr5  | 11491 |\n",
       "| chr6  | 11406 |\n",
       "| chr7  | 11618 |\n",
       "| chr8  |  8895 |\n",
       "| chr9  |  8590 |\n",
       "| chrX  |  6772 |\n",
       "| chrY  |   168 |\n",
       "\n"
      ],
      "text/plain": [
       "   Var1  Freq \n",
       "1  chr1  20635\n",
       "2  chr10  9951\n",
       "3  chr11 10892\n",
       "4  chr12 10824\n",
       "5  chr13  5165\n",
       "6  chr14  6768\n",
       "7  chr15  7403\n",
       "8  chr16  7514\n",
       "9  chr17 10333\n",
       "10 chr18  4199\n",
       "11 chr19  8918\n",
       "12 chr2  16139\n",
       "13 chr20  5466\n",
       "14 chr21  2654\n",
       "15 chr22  4405\n",
       "16 chr3  13532\n",
       "17 chr4   9908\n",
       "18 chr5  11491\n",
       "19 chr6  11406\n",
       "20 chr7  11618\n",
       "21 chr8   8895\n",
       "22 chr9   8590\n",
       "23 chrX   6772\n",
       "24 chrY    168"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.data.frame(table(gene_bidir_tpm_intragenic_B$chrom))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 24 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Var1</th><th scope=col>Freq</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1 </td><td>20521</td></tr>\n",
       "\t<tr><td>chr10</td><td>10290</td></tr>\n",
       "\t<tr><td>chr11</td><td>10703</td></tr>\n",
       "\t<tr><td>chr12</td><td>10775</td></tr>\n",
       "\t<tr><td>chr13</td><td> 5137</td></tr>\n",
       "\t<tr><td>chr14</td><td> 6729</td></tr>\n",
       "\t<tr><td>chr15</td><td> 7535</td></tr>\n",
       "\t<tr><td>chr16</td><td> 7385</td></tr>\n",
       "\t<tr><td>chr17</td><td>10243</td></tr>\n",
       "\t<tr><td>chr18</td><td> 4234</td></tr>\n",
       "\t<tr><td>chr19</td><td> 8698</td></tr>\n",
       "\t<tr><td>chr2 </td><td>16107</td></tr>\n",
       "\t<tr><td>chr20</td><td> 5349</td></tr>\n",
       "\t<tr><td>chr21</td><td> 2745</td></tr>\n",
       "\t<tr><td>chr22</td><td> 4441</td></tr>\n",
       "\t<tr><td>chr3 </td><td>13753</td></tr>\n",
       "\t<tr><td>chr4 </td><td>10068</td></tr>\n",
       "\t<tr><td>chr5 </td><td>11456</td></tr>\n",
       "\t<tr><td>chr6 </td><td>11426</td></tr>\n",
       "\t<tr><td>chr7 </td><td>11372</td></tr>\n",
       "\t<tr><td>chr8 </td><td> 8956</td></tr>\n",
       "\t<tr><td>chr9 </td><td> 8674</td></tr>\n",
       "\t<tr><td>chrX </td><td> 6802</td></tr>\n",
       "\t<tr><td>chrY </td><td>  168</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 24 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Var1 & Freq\\\\\n",
       " <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1  & 20521\\\\\n",
       "\t chr10 & 10290\\\\\n",
       "\t chr11 & 10703\\\\\n",
       "\t chr12 & 10775\\\\\n",
       "\t chr13 &  5137\\\\\n",
       "\t chr14 &  6729\\\\\n",
       "\t chr15 &  7535\\\\\n",
       "\t chr16 &  7385\\\\\n",
       "\t chr17 & 10243\\\\\n",
       "\t chr18 &  4234\\\\\n",
       "\t chr19 &  8698\\\\\n",
       "\t chr2  & 16107\\\\\n",
       "\t chr20 &  5349\\\\\n",
       "\t chr21 &  2745\\\\\n",
       "\t chr22 &  4441\\\\\n",
       "\t chr3  & 13753\\\\\n",
       "\t chr4  & 10068\\\\\n",
       "\t chr5  & 11456\\\\\n",
       "\t chr6  & 11426\\\\\n",
       "\t chr7  & 11372\\\\\n",
       "\t chr8  &  8956\\\\\n",
       "\t chr9  &  8674\\\\\n",
       "\t chrX  &  6802\\\\\n",
       "\t chrY  &   168\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 24 × 2\n",
       "\n",
       "| Var1 &lt;fct&gt; | Freq &lt;int&gt; |\n",
       "|---|---|\n",
       "| chr1  | 20521 |\n",
       "| chr10 | 10290 |\n",
       "| chr11 | 10703 |\n",
       "| chr12 | 10775 |\n",
       "| chr13 |  5137 |\n",
       "| chr14 |  6729 |\n",
       "| chr15 |  7535 |\n",
       "| chr16 |  7385 |\n",
       "| chr17 | 10243 |\n",
       "| chr18 |  4234 |\n",
       "| chr19 |  8698 |\n",
       "| chr2  | 16107 |\n",
       "| chr20 |  5349 |\n",
       "| chr21 |  2745 |\n",
       "| chr22 |  4441 |\n",
       "| chr3  | 13753 |\n",
       "| chr4  | 10068 |\n",
       "| chr5  | 11456 |\n",
       "| chr6  | 11426 |\n",
       "| chr7  | 11372 |\n",
       "| chr8  |  8956 |\n",
       "| chr9  |  8674 |\n",
       "| chrX  |  6802 |\n",
       "| chrY  |   168 |\n",
       "\n"
      ],
      "text/plain": [
       "   Var1  Freq \n",
       "1  chr1  20521\n",
       "2  chr10 10290\n",
       "3  chr11 10703\n",
       "4  chr12 10775\n",
       "5  chr13  5137\n",
       "6  chr14  6729\n",
       "7  chr15  7535\n",
       "8  chr16  7385\n",
       "9  chr17 10243\n",
       "10 chr18  4234\n",
       "11 chr19  8698\n",
       "12 chr2  16107\n",
       "13 chr20  5349\n",
       "14 chr21  2745\n",
       "15 chr22  4441\n",
       "16 chr3  13753\n",
       "17 chr4  10068\n",
       "18 chr5  11456\n",
       "19 chr6  11426\n",
       "20 chr7  11372\n",
       "21 chr8   8956\n",
       "22 chr9   8674\n",
       "23 chrX   6802\n",
       "24 chrY    168"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "as.data.frame(table(gene_bidir_tpm_intragenic_C$chrom))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Session information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.0 (2019-04-26)\n",
       "Platform: x86_64-redhat-linux-gnu (64-bit)\n",
       "Running under: CentOS Linux 7 (Core)\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] data.table_1.14.2\n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] fansi_1.0.3       crayon_1.5.1      digest_0.6.29     utf8_1.2.2       \n",
       " [5] R.methodsS3_1.8.2 IRdisplay_1.1     repr_1.1.4        lifecycle_1.0.3  \n",
       " [9] jsonlite_1.8.0    evaluate_0.16     pillar_1.8.1      rlang_1.0.6      \n",
       "[13] cli_3.4.1         uuid_1.1-0        R.oo_1.25.0       R.utils_2.12.0   \n",
       "[17] vctrs_0.5.0       IRkernel_1.3.2    tools_3.6.0       glue_1.6.2       \n",
       "[21] fastmap_1.1.0     compiler_3.6.0    base64enc_0.1-3   pbdZMQ_0.3-7     \n",
       "[25] htmltools_0.5.2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.4.0"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "283px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
